diff -ur vendor.orig/bjeavons/zxcvbn-php/src/Matchers/Bruteforce.php vendor/bjeavons/zxcvbn-php/src/Matchers/Bruteforce.php
--- vendor.orig/bjeavons/zxcvbn-php/src/Matchers/Bruteforce.php	2018-07-23 05:05:36.000000000 +0000
+++ vendor/bjeavons/zxcvbn-php/src/Matchers/Bruteforce.php	2022-01-12 15:46:47.142260962 +0000
@@ -5,7 +5,7 @@
 /**
  * Class Bruteforce.
  */
-class Bruteforce extends Match
+class Bruteforce extends BaseMatch
 {
     /**
      * @param $password
diff -ur vendor.orig/bjeavons/zxcvbn-php/src/Matchers/DateMatch.php vendor/bjeavons/zxcvbn-php/src/Matchers/DateMatch.php
--- vendor.orig/bjeavons/zxcvbn-php/src/Matchers/DateMatch.php	2018-07-23 05:05:36.000000000 +0000
+++ vendor/bjeavons/zxcvbn-php/src/Matchers/DateMatch.php	2022-01-12 15:46:47.115260914 +0000
@@ -2,7 +2,7 @@
 
 namespace ZxcvbnPhp\Matchers;
 
-class DateMatch extends Match
+class DateMatch extends BaseMatch
 {
     const NUM_YEARS = 119; // Years match against 1900 - 2019
     const NUM_MONTHS = 12;
diff -ur vendor.orig/bjeavons/zxcvbn-php/src/Matchers/DictionaryMatch.php vendor/bjeavons/zxcvbn-php/src/Matchers/DictionaryMatch.php
--- vendor.orig/bjeavons/zxcvbn-php/src/Matchers/DictionaryMatch.php	2018-07-23 05:05:36.000000000 +0000
+++ vendor/bjeavons/zxcvbn-php/src/Matchers/DictionaryMatch.php	2022-01-12 15:46:47.151260978 +0000
@@ -2,7 +2,7 @@
 
 namespace ZxcvbnPhp\Matchers;
 
-class DictionaryMatch extends Match
+class DictionaryMatch extends BaseMatch
 {
     /**
      * @var
diff -ur vendor.orig/bjeavons/zxcvbn-php/src/Matchers/DigitMatch.php vendor/bjeavons/zxcvbn-php/src/Matchers/DigitMatch.php
--- vendor.orig/bjeavons/zxcvbn-php/src/Matchers/DigitMatch.php	2018-07-23 05:05:36.000000000 +0000
+++ vendor/bjeavons/zxcvbn-php/src/Matchers/DigitMatch.php	2022-01-12 15:46:47.137260953 +0000
@@ -2,7 +2,7 @@
 
 namespace ZxcvbnPhp\Matchers;
 
-class DigitMatch extends Match
+class DigitMatch extends BaseMatch
 {
     /**
      * @param $password
diff -ur vendor.orig/bjeavons/zxcvbn-php/src/Matchers/Match.php vendor/bjeavons/zxcvbn-php/src/Matchers/Match.php
--- vendor.orig/bjeavons/zxcvbn-php/src/Matchers/Match.php	2022-01-12 15:45:58.544173708 +0000
+++ vendor/bjeavons/zxcvbn-php/src/Matchers/Match.php	2022-01-12 15:46:17.640208001 +0000
@@ -2,7 +2,7 @@
 
 namespace ZxcvbnPhp\Matchers;
 
-abstract class Match implements MatchInterface
+abstract class BaseMatch implements MatchInterface
 {
     /**
      * @var
diff -ur vendor.orig/bjeavons/zxcvbn-php/src/Matchers/RepeatMatch.php vendor/bjeavons/zxcvbn-php/src/Matchers/RepeatMatch.php
--- vendor.orig/bjeavons/zxcvbn-php/src/Matchers/RepeatMatch.php	2018-07-23 05:05:36.000000000 +0000
+++ vendor/bjeavons/zxcvbn-php/src/Matchers/RepeatMatch.php	2022-01-12 15:46:47.146260969 +0000
@@ -2,7 +2,7 @@
 
 namespace ZxcvbnPhp\Matchers;
 
-class RepeatMatch extends Match
+class RepeatMatch extends BaseMatch
 {
     /**
      * @var
diff -ur vendor.orig/bjeavons/zxcvbn-php/src/Matchers/SequenceMatch.php vendor/bjeavons/zxcvbn-php/src/Matchers/SequenceMatch.php
--- vendor.orig/bjeavons/zxcvbn-php/src/Matchers/SequenceMatch.php	2018-07-23 05:05:36.000000000 +0000
+++ vendor/bjeavons/zxcvbn-php/src/Matchers/SequenceMatch.php	2022-01-12 15:46:47.156260987 +0000
@@ -2,7 +2,7 @@
 
 namespace ZxcvbnPhp\Matchers;
 
-class SequenceMatch extends Match
+class SequenceMatch extends BaseMatch
 {
     const LOWER = 'abcdefghijklmnopqrstuvwxyz';
     const UPPER = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
diff -ur vendor.orig/bjeavons/zxcvbn-php/src/Matchers/SpatialMatch.php vendor/bjeavons/zxcvbn-php/src/Matchers/SpatialMatch.php
--- vendor.orig/bjeavons/zxcvbn-php/src/Matchers/SpatialMatch.php	2018-07-23 05:05:36.000000000 +0000
+++ vendor/bjeavons/zxcvbn-php/src/Matchers/SpatialMatch.php	2022-01-12 15:46:47.125260932 +0000
@@ -5,7 +5,7 @@
 /**
  * Class SpatialMatch.
  */
-class SpatialMatch extends Match
+class SpatialMatch extends BaseMatch
 {
     /**
      * @var
diff -ur vendor.orig/bjeavons/zxcvbn-php/src/Matchers/YearMatch.php vendor/bjeavons/zxcvbn-php/src/Matchers/YearMatch.php
--- vendor.orig/bjeavons/zxcvbn-php/src/Matchers/YearMatch.php	2018-07-23 05:05:36.000000000 +0000
+++ vendor/bjeavons/zxcvbn-php/src/Matchers/YearMatch.php	2022-01-12 15:46:47.132260944 +0000
@@ -2,7 +2,7 @@
 
 namespace ZxcvbnPhp\Matchers;
 
-class YearMatch extends Match
+class YearMatch extends BaseMatch
 {
     const NUM_YEARS = 119;
 
Only in vendor/bjeavons/zxcvbn-php: src.old
