--- core/modules/locale/locale.compare.inc	2017-09-26 10:33:39.319052548 +0000
+++ core/modules/locale/locale.compare.inc	2017-09-26 10:30:43.836862545 +0000
@@ -108,6 +108,9 @@
     $project_info = new ProjectInfo();
     $project_info->processInfoList($projects, $module_data, 'module', TRUE, $additional_whitelist);
     $project_info->processInfoList($projects, $theme_data, 'theme', TRUE, $additional_whitelist);
+    # Also add disabled modules in the list of projects to install
+    $project_info->processInfoList($projects, $module_data, 'module', FALSE, $additional_whitelist);
+    $project_info->processInfoList($projects, $theme_data, 'theme', FALSE, $additional_whitelist);
 
     // Allow other modules to alter projects before fetching and comparing.
     \Drupal::moduleHandler()->alter('locale_translation_projects', $projects);
@@ -192,7 +192,7 @@
   }
   else {
     // Retrieve and save the status of local translations only.
-    locale_translation_check_projects_local($projects, $langcodes);
+    locale_translation_check_projects_batch($projects, $langcodes);
     \Drupal::state()->set('locale.translation_last_checked', REQUEST_TIME);
   }
 }
@@ -310,6 +310,14 @@
       $source = locale_translation_source_build($project, $langcode);
       $file = locale_translation_source_check_file($source);
       locale_translation_status_save($name, $langcode, LOCALE_TRANSLATION_LOCAL, $file);
+
+      if($file === FALSE) {
+        // try sites/all/translations as well
+        $source->files['local']->directory = "sites/all/translations";
+        $source->files['local']->uri = $source->files['local']->directory . $source->files['local']->filename;
+        $file = locale_translation_source_check_file($source);
+        locale_translation_status_save($name, $langcode, LOCALE_TRANSLATION_LOCAL, $file);
+      }
     }
   }
 }
