--- core/lib/Drupal/Core/Menu/MenuTreeStorage.php       2024-03-06 08:23:56
+++ core/lib/Drupal/Core/Menu/MenuTreeStorage.php  2024-03-19 11:50:05
@@ -1224,7 +1224,7 @@
         'route_param_key' => [
           'description' => 'An encoded string of route parameters for loading by route.',
           'type' => 'varchar',
-          'length' => 255,
+          'length' => 2048,
         ],
         'route_parameters' => [
           'description' => 'Serialized array of route parameters of this menu link.',
@@ -1236,7 +1236,7 @@
         'url' => [
           'description' => 'The external path this link points to (when not using a route).',
           'type' => 'varchar',
-          'length' => 255,
+          'length' => 2048,
           'not null' => TRUE,
           'default' => '',
         ],
