--- core/modules/menu_ui/menu_ui.module	2018-11-07 19:36:14.000000000 +0000
+++ core/modules/menu_ui/menu_ui.module	2018-11-07 20:16:24.000000000 +0000
@@ -8,6 +8,7 @@
 use Drupal\menu_link_content\Entity\MenuLinkContent;
 use Drupal\node\NodeTypeInterface;
 use Drupal\node\NodeInterface;
+use Drupal\Core\Language\LanguageInterface;

 /**
  * Helper function to create or update a menu link for a node.
@@ -66,6 +67,7 @@
   /** @var \Drupal\node\NodeTypeInterface $node_type */
   $node_type = $node->type->entity;
   $menu_name = strtok($node_type->getThirdPartySetting('menu_ui', 'parent', 'main:'), ':');
+  $langcode = $node->language()->getId();
   $defaults = FALSE;
   if ($node->id()) {
     $id = FALSE;
@@ -76,6 +78,7 @@
         ->accessCheck(TRUE)
         ->condition('link.uri', 'entity:node/' . $node->id())
         ->condition('menu_name', $menu_name)
+        ->condition('langcode', array($langcode, LanguageInterface::LANGCODE_NOT_SPECIFIED, LanguageInterface::LANGCODE_NOT_APPLICABLE), 'IN')
         ->sort('id', 'ASC')
         ->range(0, 1);
       $result = $query->execute();
@@ -88,6 +91,7 @@
         ->accessCheck(TRUE)
         ->condition('link.uri', 'entity:node/' . $node->id())
         ->condition('menu_name', array_values($type_menus), 'IN')
+        ->condition('langcode', array($langcode, LanguageInterface::LANGCODE_NOT_SPECIFIED, LanguageInterface::LANGCODE_NOT_APPLICABLE), 'IN')
         ->sort('id', 'ASC')
         ->range(0, 1);
       $result = $query->execute();
