--- core/lib/Drupal/Core/DrupalKernel.php	2017-07-05 17:03:07.000000000 +0000
+++ core/lib/Drupal/Core/DrupalKernel.php	2017-07-12 20:18:52.986335783 +0000
@@ -1096,6 +1096,12 @@
 
     // Create base URL.
     $base_root = $request->getSchemeAndHttpHost();
+
+    // APIC change for subdirectories
+    global $_SERVER;
+    if (isset($_SERVER['RAW_HOST']) && !empty($_SERVER['RAW_HOST'])) {
+      $base_root = 'https://' . $_SERVER['RAW_HOST'];
+    }
     $base_url = $base_root;
 
     // For a request URI of '/index.php/foo', $_SERVER['SCRIPT_NAME'] is
@@ -1599,3 +1605,4 @@
   }
 
 }
+
