--- core/lib/Drupal/Core/Theme/ThemeManager.php	2022-06-21 21:53:48.000000000 +0100
+++ core/lib/Drupal/Core/Theme/ThemeManager.php	2022-06-29 10:11:50.000000000 +0100
@@ -284,7 +284,7 @@
     if (isset($info['preprocess functions'])) {
       foreach ($info['preprocess functions'] as $preprocessor_function) {
         if (is_callable($preprocessor_function)) {
-          call_user_func_array($preprocessor_function, [&$variables, $hook, $info]);
+          call_user_func_array($preprocessor_function, [&$variables, $hook, &$info]);
         }
       }
       // Allow theme preprocess functions to set $variables['#attached'] and
