--- core/modules/views/src/Plugin/views/field/EntityOperations.php
+++ core/modules/views/src/Plugin/views/field/EntityOperations.php
@@ -139,22 +139,24 @@ public function buildOptionsForm(&$form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function render(ResultRow $values) {
-    $entity = $this->getEntityTranslation($this->getEntity($values), $values);
-    $operations = $this->entityTypeManager->getListBuilder($entity->getEntityTypeId())->getOperations($entity);
-    if ($this->options['destination']) {
-      foreach ($operations as &$operation) {
-        if (!isset($operation['query'])) {
-          $operation['query'] = [];
+    if (isset($values->_entity)) {
+      $entity = $this->getEntityTranslation($this->getEntity($values), $values);
+      $operations = $this->entityTypeManager->getListBuilder($entity->getEntityTypeId())->getOperations($entity);
+      if ($this->options['destination']) {
+        foreach ($operations as &$operation) {
+          if (!isset($operation['query'])) {
+            $operation['query'] = [];
+          }
+          $operation['query'] += $this->getDestinationArray();
         }
-        $operation['query'] += $this->getDestinationArray();
       }
-    }
-    $build = [
-      '#type' => 'operations',
-      '#links' => $operations,
-    ];
+      $build = [
+        '#type' => 'operations',
+        '#links' => $operations,
+      ];

-    return $build;
+      return $build;
+    }
   }

   /**