--- core/lib/Drupal/Core/File/FileSystem.php    2023-07-21 12:33:09
+++ core/lib/Drupal/Core/File/FileSystem.php       2023-07-21 12:33:07
@@ -336,7 +336,7 @@
     // Return TRUE for non-existent file, but log that nothing was actually
     // deleted, as the current state is the intended result.
     if (!file_exists($path)) {
-      $this->logger->notice('The file %path was not deleted because it does not exist.', ['%path' => $path]);
+      //$this->logger->notice('The file %path was not deleted because it does not exist.', ['%path' => $path]);
       return TRUE;
     }
