--- core/modules/locale/src/Form/ImportForm.php	2019-05-09 08:49:36.000000000 +0100
+++ core/modules/locale/src/Form/ImportForm.php	2019-05-09 08:50:40.000000000 +0100
@@ -98,8 +98,16 @@
       ];
     }
 
+    // Patched by IBM - determine which languages are enabled and whitelist
+    // those filename "extensions" too
+    $enabled_languages = \Drupal::languageManager()->getLanguages();
+    $valid_extensions = 'po';
+    foreach(array_keys($enabled_languages) as $language_code){
+      $valid_extensions .= " $language_code";
+    }
+
     $validators = [
-      'file_validate_extensions' => ['po'],
+      'file_validate_extensions' => [$valid_extensions],
       'file_validate_size' => [Environment::getUploadMaxSize()],
     ];
     $form['file'] = [
