--- core/modules/locale/locale.batch.inc	2017-10-06 13:35:46.254378029 +0000
+++ core/modules/locale/locale.batch.inc	2017-10-06 13:34:13.178164353 +0000
@@ -48,9 +48,20 @@
 
   // Check the status of local translation files.
   if (isset($source->files[LOCALE_TRANSLATION_LOCAL])) {
-    if ($file = locale_translation_source_check_file($source)) {
+    $file = locale_translation_source_check_file($source);
+    if ($file !== FALSE) {
       locale_translation_status_save($source->name, $source->langcode, LOCALE_TRANSLATION_LOCAL, $file);
     }
+    else {
+      // try sites/all/translations as well
+      $source->files['local']->directory = "sites/all/translations";
+      $source->files['local']->uri = $source->files['local']->directory . $source->files['local']->filename;
+      $file = locale_translation_source_check_file($source);
+      if ($file !== FALSE) {
+        locale_translation_status_save($source->name, $source->langcode, LOCALE_TRANSLATION_LOCAL, $file);
+      }
+    }
+
     $checked = TRUE;
   }
