--- core/lib/Drupal/Core/Render/MetadataBubblingUrlGenerator.php        2024-09-03 14:15:15
+++ core/lib/Drupal/Core/Render/MetadataBubblingUrlGenerator.php   2024-09-03 14:14:58
@@ -109,4 +109,11 @@
     return $collect_bubbleable_metadata ? $generated_url : $generated_url->getGeneratedUrl();
   }

+  /**
+   * {@inheritdoc}
+   */
+  public function generateActualFromRoute($name, $parameters = [], $options = []) {
+    return $this->urlGenerator->generateFromRoute($name, $parameters, $options, FALSE);
+  }
+
 }
