--- core/lib/Drupal/Core/Utility/UnroutedUrlAssembler.php	2017-09-22 16:18:54.000000000 +0100
+++ core/lib/Drupal/Core/Utility/UnroutedUrlAssembler.php	2017-09-26 13:54:57.084776104 +0100
@@ -165,6 +165,12 @@
     $uri = $options['script'] && ($uri != '') ? '/' . trim($uri, '/') : $uri;
     $query = $options['query'] ? ('?' . UrlHelper::buildQuery($options['query'])) : '';
     $url = $base . $options['script'] . $uri . $query . $options['fragment'];
+
+    // Added by IBM to allow module installs to work properly in a multisite environment
+    if (isset($GLOBALS['real_base_url']) && isset($GLOBALS['real_base_host']) && $GLOBALS['real_base_url'] != $GLOBALS['real_base_host']) {
+      $url = str_replace($base, $GLOBALS['real_base_url'] . "/", $url);
+    }
+
     return $collect_bubbleable_metadata ? $generated_url->setGeneratedUrl($url) : $url;
   }

