--- core/modules/user/src/Form/UserPasswordForm.php	2018-03-27 11:47:10.240128098 +0000
+++ core/modules/user/src/Form/UserPasswordForm.php	2018-03-27 11:47:46.808061754 +0000
@@ -226,7 +226,7 @@
     // Make sure the status text is displayed even if no email was sent. This
     // message is deliberately the same as the success message for privacy.
     $this->messenger()
-      ->addStatus($this->t('If %identifier is a valid account, an email will be sent with instructions to reset your password.', [
+      ->addStatus($this->t('If the account exists, an email has been sent with further instructions to reset the password.', [
         '%identifier' => $form_state->getValue('name'),
       ]));
 
