--- core/modules/system/system.install  2024-03-06 08:23:56
+++ core/modules/system/system.install     2024-03-19 11:54:45
@@ -1948,4 +1948,26 @@
   }

   $sandbox['#finished'] = $sandbox['current_key'] == $sandbox['max'];
+}
+
+/**
+ * Update length of menu_tree fields url and route_param_key from 255 to 2048.
+ */
+function system_update_10202() {
+  $schema = \Drupal::database()->schema();
+  $spec = [
+    'description' => 'The external path this link points to (when not using a route).',
+    'type' => 'varchar',
+    'length' => 2048,
+    'not null' => TRUE,
+    'default' => '',
+  ];
+  $schema->changeField('menu_tree', 'url', 'url', $spec);
+
+  $spec = [
+    'description' => 'An encoded string of route parameters for loading by route.',
+    'type' => 'varchar',
+    'length' => 2048,
+  ];
+  $schema->changeField('menu_tree', 'route_param_key', 'route_param_key', $spec);
 }
