--- core/lib/Drupal/Core/Password/PhpassHashedPasswordBase.php	2024-07-04 12:33:45.000000000 +0100
+++ core/lib/Drupal/Core/Password/PhpassHashedPasswordBase.php	2024-07-16 13:50:43.569058199 +0100
@@ -217,9 +217,9 @@
     // Convert the base 2 logarithm into an integer.
     $count = 1 << $count_log2;
 
-    $hash = hash($algo, $salt . $password, TRUE);
+    $hash = openssl_digest($salt . $password, $algo, TRUE);
     do {
-      $hash = hash($algo, $hash . $password, TRUE);
+      $hash = openssl_digest($hash . $password, $algo, TRUE);
     } while (--$count);
 
     $len = strlen($hash);
