--- core/modules/forum/src/Controller/ForumController.php	2018-01-24 14:25:02.000000000 +0000
+++ core/modules/forum/src/Controller/ForumController.php	2018-01-24 14:28:41.000000000 +0000
@@ -330,7 +330,12 @@
             'url' => Url::fromRoute('user.login', [], ['query' => $this->getDestinationArray()]),
           ],
         ];
+        $links['#prefix'] = '<ul class="forum-actions-wrapper">';
+        $links['#suffix'] = '</ul>';
       }
+    } else {
+      $links['#prefix'] = '<ul class="forum-actions-wrapper">';
+      $links['#suffix'] = '</ul>';
     }
     return $links;
   }
