--- core/modules/user/src/Form/UserLoginForm.php	2017-10-16 11:06:29.000000000 +0100
+++ core/modules/user/src/Form/UserLoginForm.php	2017-10-16 11:04:44.000000000 +0100
@@ -149,7 +149,10 @@
   public function validateName(array &$form, FormStateInterface $form_state) {
     if (!$form_state->isValueEmpty('name') && user_is_blocked($form_state->getValue('name'))) {
       // Blocked in user administration.
-      $form_state->setErrorByName('name', $this->t('The username %name has not been activated or is blocked.', ['%name' => $form_state->getValue('name')]));
+      //$form_state->setErrorByName('name', $this->t('The username %name has not been activated or is blocked.', ['%name' => $form_state->getValue('name')]));
+      $this->logger('user')->notice('Login attempt for %user which has not been activated or is blocked.', ['%user' => $form_state->getValue('name')]);
+      $form_state->setErrorByName('usernameorpassword', $this->t('The credentials provided for authentication are invalid. Please repeat the request with valid credentials. Please note that repeated attempts with incorrect credentials can lock the user account.'));
+      $form_state->setErrorByName('usernameorpassword2', $this->t('Unrecognized username or password. <a href=":password">Forgot your password?</a>', [':password' => $this->url('user.pass', [], ['query' => $query])]));
     }
   }
 
@@ -218,11 +221,17 @@
 
       if ($flood_control_triggered = $form_state->get('flood_control_triggered')) {
         if ($flood_control_triggered == 'user') {
-          $form_state->setErrorByName('name', $this->formatPlural($flood_config->get('user_limit'), 'There has been more than one failed login attempt for this account. It is temporarily blocked. Try again later or <a href=":url">request a new password</a>.', 'There have been more than @count failed login attempts for this account. It is temporarily blocked. Try again later or <a href=":url">request a new password</a>.', [':url' => $this->url('user.pass')]));
+          //$form_state->setErrorByName('name', $this->formatPlural($flood_config->get('user_limit'), 'There has been more than one failed login attempt for this account. It is temporarily blocked. Try again later or <a href=":url">request a new password</a>.', 'There have been more than @count failed login attempts for this account. It is temporarily blocked. Try again later or <a href=":url">request a new password</a>.', [':url' => $this->url('user.pass')]));
+          $this->logger('user')->notice('Login attempt blocked for %user.', ['%user' => $form_state->getValue('name')]);
+          $form_state->setErrorByName('usernameorpassword', $this->t('The credentials provided for authentication are invalid. Please repeat the request with valid credentials. Please note that repeated attempts with incorrect credentials can lock the user account.'));
+          $form_state->setErrorByName('usernameorpassword2', $this->t('Unrecognized username or password. <a href=":password">Forgot your password?</a>', [':password' => $this->url('user.pass', [], ['query' => $query])]));
         }
         else {
           // We did not find a uid, so the limit is IP-based.
-          $form_state->setErrorByName('name', $this->t('Too many failed login attempts from your IP address. This IP address is temporarily blocked. Try again later or <a href=":url">request a new password</a>.', [':url' => $this->url('user.pass')]));
+          //$form_state->setErrorByName('name', $this->t('Too many failed login attempts from your IP address. This IP address is temporarily blocked. Try again later or <a href=":url">request a new password</a>.', [':url' => $this->url('user.pass')]));
+          $this->logger('user')->notice('Login attempt blocked from %ip.', ['%ip' => $this->getRequest()->getClientIp()]);
+          $form_state->setErrorByName('usernameorpassword', $this->t('The credentials provided for authentication are invalid. Please repeat the request with valid credentials. Please note that repeated attempts with incorrect credentials can lock the user account.'));
+          $form_state->setErrorByName('usernameorpassword2', $this->t('Unrecognized username or password. <a href=":password">Forgot your password?</a>', [':password' => $this->url('user.pass', [], ['query' => $query])]));
         }
       }
       else {
@@ -232,7 +241,8 @@
         // handlers that ran earlier than this one.
         $user_input = $form_state->getUserInput();
         $query = isset($user_input['name']) ? ['name' => $user_input['name']] : [];
-        $form_state->setErrorByName('name', $this->t('Unrecognized username or password. <a href=":password">Forgot your password?</a>', [':password' => $this->url('user.pass', [], ['query' => $query])]));
+        $form_state->setErrorByName('usernameorpassword', $this->t('The credentials provided for authentication are invalid. Please repeat the request with valid credentials. Please note that repeated attempts with incorrect credentials can lock the user account.'));
+        $form_state->setErrorByName('usernameorpassword2', $this->t('Unrecognized username or password. <a href=":password">Forgot your password?</a>', [':password' => $this->url('user.pass', [], ['query' => $query])]));
         $accounts = $this->userStorage->loadByProperties(['name' => $form_state->getValue('name')]);
         if (!empty($accounts)) {
           $this->logger('user')->notice('Login attempt failed for %user.', ['%user' => $form_state->getValue('name')]);
