--- core/lib/Drupal/Core/Extension/Extension.php	2022-11-17 12:25:02
+++ core/lib/Drupal/Core/Extension/Extension.php	2022-11-17 12:24:55
@@ -156,10 +156,17 @@
    * Offers all SplFileInfo methods to consumers; e.g., $extension->getMTime().
    */
   public function __call($method, array $args) {
+    $result = null;
     if (!isset($this->splFileInfo)) {
-      $this->splFileInfo = new \SplFileInfo($this->root . '/' . $this->pathname);
+      try {
+        $this->splFileInfo = new \SplFileInfo($this->root . '/' . $this->pathname);
+        $result = call_user_func_array([$this->splFileInfo, $method], $args);
+      } catch(\Throwable $e) {
+        $this->splFileInfo = new \SplFileInfo($this->pathname);
+        $result = call_user_func_array([$this->splFileInfo, $method], $args);
+      }
     }
-    return call_user_func_array([$this->splFileInfo, $method], $args);
+    return $result;
   }

   /**
