--- core/modules/user/src/Hook/UserHooks.php    2025-01-08 11:49:20
+++ core/modules/user/src/Hook/UserHooks.php       2025-01-08 11:48:59
@@ -310,8 +310,26 @@
       ]);
       $action->trustData()->save();
     }
+  }
+
+/**
+ * Implements hook_ENTITY_TYPE_update() for user_role entities.
+ */
+#[Hook('user_role_update')]
+function userRoleUpdate(RoleInterface $role) {
+  $add_action = Action::load('user_add_role_action.' . $role->id());
+  if ($add_action) {
+    $add_action->set('label', t('Add the @label role to the selected user(s)', ['@label' => $role->label()]));
+    $add_action->save();
   }

+  $remove_action = Action::load('user_remove_role_action.' . $role->id());
+  if ($remove_action) {
+    $remove_action->set('label', t('Remove the @label role from the selected user(s)', ['@label' => $role->label()]));
+    $remove_action->save();
+  }
+}
+
   /**
    * Implements hook_ENTITY_TYPE_delete() for user_role entities.
    */