--- core/lib/Drupal/Core/Routing/LocalAwareRedirectResponseTrait.php	2017-01-18 22:03:15.000000000 +0000
+++ core/lib/Drupal/Core/Routing/LocalAwareRedirectResponseTrait.php	2017-01-20 08:42:05.000000000 +0000
@@ -20,7 +20,12 @@
    * {@inheritdoc}
    */
   protected function isLocal($url) {
-    return !UrlHelper::isExternal($url) || UrlHelper::externalIsLocal($url, $this->getRequestContext()->getCompleteBaseUrl());
+    //return !UrlHelper::isExternal($url) || UrlHelper::externalIsLocal($url, $this->getRequestContext()->getCompleteBaseUrl());
+    if (isset($GLOBALS['real_base_url'])) {
+      return !UrlHelper::isExternal($url) || UrlHelper::externalIsLocal($url, $GLOBALS['real_base_url']);
+    } else {
+      return !UrlHelper::isExternal($url) || UrlHelper::externalIsLocal($url, $this->getRequestContext()->getCompleteBaseUrl());
+    }
   }
 
   /**
