--- core/modules/views/src/Plugin/views/display/PathPluginBase.php
+++ core/modules/views/src/Plugin/views/display/PathPluginBase.php
@@ -507,6 +507,10 @@ protected function validatePath($path) {
       $errors[] = $this->t('Invalid path. Valid characters are alphanumerics as well as "-", ".", "_" and "~".');
     }
 
+    if (UrlHelper::isExternal($path)) {
+      $errors[] = $this->t("Path component '{$path}' is external. External URL is not accepted as view path.");
+    }
+
     $path_sections = explode('/', $path);
     // Symfony routing does not allow to use numeric placeholders.
     // @see \Symfony\Component\Routing\RouteCompiler