--- core/lib/Drupal/Component/Gettext/PoItem.php	2017-08-15 12:16:59.969245368 +0000
+++ core/lib/Drupal/Component/Gettext/PoItem.php	2017-08-15 12:17:57.621967647 +0000
@@ -264,9 +264,11 @@
     $string = addcslashes($string, "\0..\37\\\"");
 
     // Always include a line break after the explicit \n line breaks from
-    // the source string. Otherwise wrap at 70 chars to accommodate the extra
+    // the source string. Otherwise wrap at 700000 chars to accommodate the extra
     // format overhead too.
-    $parts = explode("\n", wordwrap(str_replace('\n', "\\n\n", $string), 70, " \n"));
+    // APIC - change line wrap to 700000 from 70 to prevent wrapping if at all possible due to
+    // limitations of IBM Translation Centers.
+    $parts = explode("\n", wordwrap(str_replace('\n', "\\n\n", $string), 700000, " \n"));
 
     // Multiline string should be exported starting with a "" and newline to
     // have all lines aligned on the same column.
