--- core/themes/seven/templates/classy/layout/html.html.twig	2021-11-19 15:04:30.076323666 +0000
+++ core/themes/seven/templates/classy/layout/html.html.twig	2021-11-19 15:26:45.844506822 +0000
@@ -44,9 +44,11 @@
       Keyboard navigation/accessibility link to main content section in
       page.html.twig.
     #}
-    <a href="#main-content" class="visually-hidden focusable skip-link">
-      {{ 'Skip to main content'|t }}
-    </a>
+    <div class="visually-hidden" role="complementary" aria-label="{{ 'Skip to main content'|t }}">
+      <a href="#main-content" class="visually-hidden focusable skip-link">
+        {{ 'Skip to main content'|t }}
+      </a>
+    </div>
     {{ page_top }}
     {{ page }}
     {{ page_bottom }}
