--- core/lib/Drupal/Core/Config/StorableConfigBase.php	2020-11-19 11:45:29.000000000 +0000
+++ core/lib/Drupal/Core/Config/StorableConfigBase.php	2020-11-19 13:06:59.000000000 +0000
@@ -180,7 +180,11 @@
   protected function castValue($key, $value) {
     $element = $this->getSchemaWrapper();
     if ($key !== NULL) {
-      $element = $element->get($key);
+      if (method_exists($element, 'get')) {
+        $element = $element->get($key);
+      } else {
+        $element = $this->get($key);
+      }
     }

     // Do not cast value if it is unknown or defined to be ignored.
