--- core/modules/config_translation/src/ConfigNamesMapper.php	2017-09-22 16:53:37.000000000 +0100
+++ core/modules/config_translation/src/ConfigNamesMapper.php	2017-09-22 17:40:00.000000000 +0100
@@ -229,16 +229,20 @@
    * {@inheritdoc}
    */
   public function getOverviewRoute() {
-    $route = new Route(
-      $this->getBaseRoute()->getPath() . '/translate',
-      [
-        '_controller' => '\Drupal\config_translation\Controller\ConfigTranslationController::itemPage',
-        'plugin_id' => $this->getPluginId(),
-      ],
-      ['_config_translation_overview_access' => 'TRUE']
-    );
-    $this->processRoute($route);
-    return $route;
+    if ($this->getBaseRoute() != null) {
+      $route = new Route(
+        $this->getBaseRoute()->getPath() . '/translate',
+        [
+          '_controller' => '\Drupal\config_translation\Controller\ConfigTranslationController::itemPage',
+          'plugin_id' => $this->getPluginId(),
+        ],
+        ['_config_translation_overview_access' => 'TRUE']
+      );
+      $this->processRoute($route);
+      return $route;
+    } else {
+      return null;
+    }
   }
 
   /**
@@ -270,16 +274,20 @@
    * {@inheritdoc}
    */
   public function getAddRoute() {
-    $route = new Route(
-      $this->getBaseRoute()->getPath() . '/translate/{langcode}/add',
-      [
-        '_form' => '\Drupal\config_translation\Form\ConfigTranslationAddForm',
-        'plugin_id' => $this->getPluginId(),
-      ],
-      ['_config_translation_form_access' => 'TRUE']
-    );
-    $this->processRoute($route);
-    return $route;
+    if ($this->getBaseRoute() != null) {
+      $route = new Route(
+        $this->getBaseRoute()->getPath() . '/translate/{langcode}/add',
+        [
+          '_form' => '\Drupal\config_translation\Form\ConfigTranslationAddForm',
+          'plugin_id' => $this->getPluginId(),
+        ],
+        ['_config_translation_form_access' => 'TRUE']
+      );
+      $this->processRoute($route);
+      return $route;
+    } else {
+      return null;
+    }
   }
 
   /**
@@ -300,16 +308,20 @@
    * {@inheritdoc}
    */
   public function getEditRoute() {
-    $route = new Route(
-      $this->getBaseRoute()->getPath() . '/translate/{langcode}/edit',
-      [
-        '_form' => '\Drupal\config_translation\Form\ConfigTranslationEditForm',
-        'plugin_id' => $this->getPluginId(),
-      ],
-      ['_config_translation_form_access' => 'TRUE']
-    );
-    $this->processRoute($route);
-    return $route;
+    if ($this->getBaseRoute() != null) {
+      $route = new Route(
+       $this->getBaseRoute()->getPath() . '/translate/{langcode}/edit',
+       [
+         '_form' => '\Drupal\config_translation\Form\ConfigTranslationEditForm',
+         'plugin_id' => $this->getPluginId(),
+        ],
+        ['_config_translation_form_access' => 'TRUE']
+      );
+      $this->processRoute($route);
+      return $route;
+    } else {
+      return null;
+    }
   }
 
   /**
@@ -330,16 +342,20 @@
    * {@inheritdoc}
    */
   public function getDeleteRoute() {
-    $route = new Route(
-      $this->getBaseRoute()->getPath() . '/translate/{langcode}/delete',
-      [
-        '_form' => '\Drupal\config_translation\Form\ConfigTranslationDeleteForm',
-        'plugin_id' => $this->getPluginId(),
-      ],
-      ['_config_translation_form_access' => 'TRUE']
-    );
-    $this->processRoute($route);
-    return $route;
+    if ($this->getBaseRoute() != null) {
+      $route = new Route(
+        $this->getBaseRoute()->getPath() . '/translate/{langcode}/delete',
+        [
+          '_form' => '\Drupal\config_translation\Form\ConfigTranslationDeleteForm',
+          'plugin_id' => $this->getPluginId(),
+        ],
+        ['_config_translation_form_access' => 'TRUE']
+      );
+      $this->processRoute($route);
+      return $route;
+    } else {
+      return null;
+    }
   }
 
   /**
