--- core/modules/update/update.module   2023-08-02 11:10:11
+++ core/modules/update/update.module   2023-08-02 15:30:32
@ -190,7 +190,7 @@
   switch ($msg_reason) {
     case UpdateManagerInterface::NOT_SECURE:
       if ($msg_type == 'core') {
-        $text = t('There is a security update available for your version of Drupal. To ensure the security of your server, you should update immediately!', [], ['langcode' => $langcode]);
+        $text = t('There are updates available for your version of Drupal. To ensure the proper functioning of your site, you should update as soon as possible.', [], ['langcode' => $langcode]);
       }
       else {
         $text = t('There are security updates available for one or more of your modules or themes. To ensure the security of your server, you should update immediately!', [], ['langcode' => $langcode]);
@@ -345,18 +345,11 @@
  * Clears the temporary files and directories based on file age from disk.
  */
 function update_clear_update_disk_cache() {
-  // List of update module cache directories. Do not create the directories if
-  // they do not exist.
-  $directories = [
-    _update_manager_cache_directory(FALSE),
-    _update_manager_extract_directory(FALSE),
-  ];
+  // Temporary directory for the site
+  $directory = "temporary://";
 
-  // Search for files and directories in base folder only without recursion.
-  foreach ($directories as $directory) {
-    if (is_dir($directory)) {
-      \Drupal::service('file_system')->scanDirectory($directory, '/.*/', ['callback' => 'update_delete_file_if_stale', 'recurse' => FALSE]);
-    }
+  if (is_dir($directory)) {
+    \Drupal::service('file_system')->scanDirectory($directory, '/.*/', ['callback' => 'update_delete_file_if_stale', 'recurse' => FALSE]);
   }
 }
