--- core/modules/big_pipe/src/Controller/BigPipeController.php  2023-09-19 13:43:38
+++ core/modules/big_pipe/src/Controller/BigPipeController.php     2023-09-19 13:43:35
@@ -52,7 +52,7 @@

     $response = new LocalRedirectResponse($request->query->get('destination'));
     // Set cookie without httpOnly, so that JavaScript can delete it.
-    $response->headers->setCookie(new Cookie(BigPipeStrategy::NOJS_COOKIE, TRUE, 0, '/', NULL, FALSE, FALSE, FALSE, NULL));
+    $response->headers->setCookie(new Cookie(BigPipeStrategy::NOJS_COOKIE, TRUE, 0, '/', NULL, TRUE, FALSE, FALSE, NULL));
     $response->addCacheableDependency((new CacheableMetadata())->addCacheContexts(['cookies:' . BigPipeStrategy::NOJS_COOKIE, 'session.exists']));
     return $response;
   }
