--- core/modules/user/src/Form/UserLoginForm.php        2024-09-03 15:28:17
+++ core/modules/user/src/Form/UserLoginForm.php   2024-09-03 15:30:18
@@ -249,6 +249,23 @@
   }

   /**
+   * Sets an error if supplied username has been blocked.
+   *
+   * Removed in Drupal 11 but added back by IBM APIC as validateName is now done in validateAuthentication
+   * We don't want this though as we have a different validation process and want to just be able to see
+   * if the name is blocked.
+   */
+  public function validateName(array &$form, FormStateInterface $form_state) {
+    if (!$form_state->isValueEmpty('name') && user_is_blocked($form_state->getValue('name'))) {
+      $user_input = $form_state->getUserInput();
+      $query = isset($user_input['name']) ? ['name' => $user_input['name']] : [];
+      $this->logger('user')->notice('Login attempt for %user which has not been activated or is blocked.', ['%user' => $form_state->getValue('name')]);
+      $form_state->setErrorByName('usernameorpassword', $this->t('Unable to sign in. This may be because the credentials provided for authentication are invalid or the user has not been activated. Please check that the user is active, then repeat the request with valid credentials. Please note that repeated attempts with incorrect credentials can lock the user account.'));
+      $form_state->setErrorByName('usernameorpassword2', $this->t('<a href=":password">Forgot your password? Click here to reset it.</a>', [':password' => Url::fromRoute('user.pass', [], ['query' => $query])->toString()]));
+    }
+  }
+
+  /**
    * Checks if user was not authenticated, or if too many logins were attempted.
    *
    * This validation function should always be the last one.
@@ -263,20 +280,24 @@
         $this->userFloodControl->register('user.failed_login_user', $flood_config->get('user_window'), $flood_control_user_identifier);
       }
 
+      $query = isset($user_input['name']) ? ['name' => $user_input['name']] : [];
       if ($flood_control_triggered = $form_state->get('flood_control_triggered')) {
+        $user_input = $form_state->getUserInput();
         if ($flood_control_triggered == 'user') {
-          $message = $this->formatPlural($flood_config->get('user_limit'), 'There has been more than one failed login attempt for this account. It is temporarily blocked. Try again later or <a href=":url">request a new password</a>.', 'There have been more than @count failed login attempts for this account. It is temporarily blocked. Try again later or <a href=":url">request a new password</a>.', [':url' => Url::fromRoute('user.pass')->toString()]);
+          $this->logger('user')->notice('Login attempt blocked for %user.', ['%user' => $form_state->getValue('name')]);
+          $form_state->setErrorByName('usernameorpassword', $this->t('Unable to sign in. This may be because the credentials provided for authentication are invalid or the user has not been activated. Please check that the user is active, then repeat the request with valid credentials. Please note that repeated attempts with incorrect credentials can lock the user account.'));
+          $form_state->setErrorByName('usernameorpassword2', $this->t('<a href=":password">Forgot your password? Click here to reset it.</a>', [':password' => Url::fromRoute('user.pass', [], ['query' => $query])->toString()]));
         }
         else {
           // We did not find a uid, so the limit is IP-based.
-          $message = $this->t('Too many failed login attempts from your IP address. This IP address is temporarily blocked. Try again later or <a href=":url">request a new password</a>.', [':url' => Url::fromRoute('user.pass')->toString()]);
+          $this->logger('user')->notice('Login attempt blocked from %ip.', ['%ip' => $this->getRequest()->getClientIp()]);
+          $form_state->setErrorByName('usernameorpassword', $this->t('Unable to sign in. This may be because the credentials provided for authentication are invalid or the user has not been activated. Please check that the user is active, then repeat the request with valid credentials. Please note that repeated attempts with incorrect credentials can lock the user account.'));
+          $form_state->setErrorByName('usernameorpassword2', $this->t('<a href=":password">Forgot your password? Click here to reset it.</a>', [':password' => Url::fromRoute('user.pass', [], ['query' => $query])->toString()]));
         }
-        $response = $this->bareHtmlPageRenderer->renderBarePage(['#markup' => $message], $this->t('Login failed'), 'maintenance_page__flood');
-        $response->setStatusCode(403);
-        $form_state->setResponse($response);
       }
       else {
-        $form_state->setErrorByName('name', $this->t('Unrecognized username or password. <a href=":password">Forgot your password?</a>', [':password' => Url::fromRoute('user.pass')->toString()]));
+        $form_state->setErrorByName('usernameorpassword', $this->t('Unable to sign in. This may be because the credentials provided for authentication are invalid or the user has not been activated. Please check that the user is active, then repeat the request with valid credentials. Please note that repeated attempts with incorrect credentials can lock the user account.'));
+        $form_state->setErrorByName('usernameorpassword2', $this->t('<a href=":password">Forgot your password? Click here to reset it.</a>', [':password' => Url::fromRoute('user.pass', [], ['query' => $query])->toString()]));
         $accounts = $this->userStorage->loadByProperties(['name' => $form_state->getValue('name')]);
         if (!empty($accounts)) {
           $this->logger('user')->notice('Login attempt failed for %user.', ['%user' => $form_state->getValue('name')]);
