--- core/modules/config_translation/src/Routing/RouteSubscriber.php	2017-09-22 17:07:11.000000000 +0100
+++ core/modules/config_translation/src/Routing/RouteSubscriber.php	2017-09-22 17:22:51.000000000 +0100
@@ -36,10 +36,18 @@
     $mappers = $this->mapperManager->getMappers($collection);
 
     foreach ($mappers as $mapper) {
-      $collection->add($mapper->getOverviewRouteName(), $mapper->getOverviewRoute());
-      $collection->add($mapper->getAddRouteName(), $mapper->getAddRoute());
-      $collection->add($mapper->getEditRouteName(), $mapper->getEditRoute());
-      $collection->add($mapper->getDeleteRouteName(), $mapper->getDeleteRoute());
+      if ($mapper->getOverviewRoute() != null) {
+        $collection->add($mapper->getOverviewRouteName(), $mapper->getOverviewRoute());
+      }
+      if ($mapper->getAddRoute() != null) {
+        $collection->add($mapper->getAddRouteName(), $mapper->getAddRoute());
+      }
+      if ($mapper->getEditRoute() != null) {
+        $collection->add($mapper->getEditRouteName(), $mapper->getEditRoute());
+      }
+      if ($mapper->getDeleteRoute() != null) {
+        $collection->add($mapper->getDeleteRouteName(), $mapper->getDeleteRoute());
+      }
     }
   }
 
