--- core/modules/user/src/Plugin/Block/UserLoginBlock.php	2022-05-23 09:20:24.154000913 +0000
+++ core/modules/user/src/Plugin/Block/UserLoginBlock.php	2022-05-23 13:30:47.085197363 +0000
@@ -99,7 +99,7 @@
    * {@inheritdoc}
    */
   public function build() {
-    $form = $this->formBuilder->getForm(UserLoginForm::class);
+    $form = $this->formBuilder->getForm('Drupal\auth_apic\Form\ApicUserLoginForm');
     unset($form['name']['#attributes']['autofocus']);
     $form['name']['#size'] = 15;
     $form['pass']['#size'] = 15;
