--- core/modules/system/src/Form/ThemeSettingsForm.php
+++ core/modules/system/src/Form/ThemeSettingsForm.php
@@ -461,9 +461,11 @@ public function submitForm(array &$form, FormStateInterface $form_state) {

     // If the user uploaded a new logo or favicon, save it to a permanent location
     // and use it in place of the default theme-provided file.
+    // Can't use private for these files as they will not be accesible.
+    $file_default_scheme = (file_default_scheme() == 'private')? 'public' : file_default_scheme();
     try {
       if (!empty($values['logo_upload'])) {
-        $filename = $this->fileSystem->copy($values['logo_upload']->getFileUri(), file_default_scheme() . '://');
+        $filename = $this->fileSystem->copy($values['logo_upload']->getFileUri(), $file_default_scheme . '://');
         $values['default_logo'] = 0;
         $values['logo_path'] = $filename;
       }
@@ -473,7 +475,7 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     }
     try {
       if (!empty($values['favicon_upload'])) {
-        $filename = $this->fileSystem->copy($values['favicon_upload']->getFileUri(), file_default_scheme() . '://');
+        $filename = $this->fileSystem->copy($values['favicon_upload']->getFileUri(), $file_default_scheme . '://');
         $values['default_favicon'] = 0;
         $values['favicon_path'] = $filename;
         $values['toggle_favicon'] = 1;