--- core/modules/update/update.module	2020-09-25 15:38:08.000000000 +0100
+++ core/modules/update/update.module	2020-09-28 08:41:13.000000000 +0100
@@ -802,18 +802,12 @@
  * Clears the temporary files and directories based on file age from disk.
  */
 function update_clear_update_disk_cache() {
-  // List of update module cache directories. Do not create the directories if
-  // they do not exist.
-  $directories = [
-    _update_manager_cache_directory(FALSE),
-    _update_manager_extract_directory(FALSE),
-  ];
+  // Temporary directory for the site
+  $directory = "temporary://";
 
   // Search for files and directories in base folder only without recursion.
-  foreach ($directories as $directory) {
-    if (is_dir($directory)) {
-      \Drupal::service('file_system')->scanDirectory($directory, '/.*/', ['callback' => 'update_delete_file_if_stale', 'recurse' => FALSE]);
-    }
+  if (is_dir($directory)) {
+    \Drupal::service('file_system')->scanDirectory($directory, '/.*/', ['callback' => 'update_delete_file_if_stale', 'recurse' => FALSE]);
   }
 }
 
