--- core/modules/system/src/Form/ThemeSettingsForm.php	2018-05-16 11:52:11.000000000 +0100
+++ core/modules/system/src/Form/ThemeSettingsForm.php	2018-05-16 11:54:12.000000000 +0100
@@ -442,12 +442,14 @@
     // If the user uploaded a new logo or favicon, save it to a permanent location
     // and use it in place of the default theme-provided file.
     if (!empty($values['logo_upload'])) {
-      $filename = file_unmanaged_copy($values['logo_upload']->getFileUri());
+      $dest = 'public://' . $values['logo_upload']->getFileName();
+      $filename = file_unmanaged_copy($values['logo_upload']->getFileUri(), $dest);
       $values['default_logo'] = 0;
       $values['logo_path'] = $filename;
     }
     if (!empty($values['favicon_upload'])) {
-      $filename = file_unmanaged_copy($values['favicon_upload']->getFileUri());
+      $dest = 'public://' . $values['favicon_upload']->getFileName();
+      $filename = file_unmanaged_copy($values['favicon_upload']->getFileUri(), $dest);
       $values['default_favicon'] = 0;
       $values['favicon_path'] = $filename;
       $values['toggle_favicon'] = 1;
