--- core/modules/system/system.module	2019-05-07 13:51:31.000000000 +0100
+++ core/modules/system/system.module	2019-05-07 13:52:39.000000000 +0100
@@ -151,7 +151,7 @@
     case 'system.modules_list':
       $output = '<p>' . t('Download additional <a href=":modules">contributed modules</a> to extend your site\'s functionality.', [':modules' => 'https://www.drupal.org/project/modules']) . '</p>';
       if (!\Drupal::moduleHandler()->moduleExists('update')) {
-        $output .= '<p>' . t('Regularly review available updates to maintain a secure and current site. Always run the <a href=":update-php">update script</a> each time a module is updated. Enable the <a href=":update-manager">Update Manager module</a> to update and install modules and themes.', [':update-php' => Url::fromRoute('system.db_update')->toString(), ':update-manager' => Url::fromRoute('system.modules_list', [], ['fragment' => 'module-update'])->toString()]) . '</p>';
+        $output .= '<p>' . t('Regularly review available updates to maintain a secure and current site. Enable the <a href=":update-manager">Update Manager module</a> to update and install modules and themes.', [':update-manager' => Url::fromRoute('system.modules_list', [], ['fragment' => 'module-update'])->toString()]) . '</p>';
       }
       return $output;
 
