--- core/lib/Drupal/Core/Routing/UrlGenerator.php	2017-08-31 11:03:37.627216844 +0000
+++ core/lib/Drupal/Core/Routing/UrlGenerator.php	2017-08-31 11:02:18.263050080 +0000
@@ -331,13 +331,16 @@
       if (isset($options['https'])) {
         if ($options['https'] === TRUE) {
           $base_url = str_replace('http://', 'https://', $base_url);
+          if (isset($GLOBALS['real_base_url']) && isset($GLOBALS['real_base_host']) && $GLOBALS['real_base_url'] != $GLOBALS['real_base_host']) $base_url = str_replace('https://', $base_url, $GLOBALS['real_base_url'], $base_url);
         }
         elseif ($options['https'] === FALSE) {
           $base_url = str_replace('https://', 'http://', $base_url);
+          if (isset($GLOBALS['real_base_url']) && isset($GLOBALS['real_base_host']) && $GLOBALS['real_base_url'] != $GLOBALS['real_base_host']) $base_url = str_replace('https://', $base_url, $GLOBALS['real_base_url'], $base_url);
         }
       }
 
       $url = $base_url . $path . $query . $fragment;
+      if (isset($GLOBALS['real_base_url']) && isset($GLOBALS['real_base_host']) && $GLOBALS['real_base_url'] != $GLOBALS['real_base_host']) $url = str_replace($base_url, $GLOBALS['real_base_url'], $url);
       return $collect_bubbleable_metadata ? $generated_url->setGeneratedUrl($url) : $url;
     }
 
@@ -372,6 +375,14 @@
       $generated_url->addCacheContexts(['url.site']);
     }
     $url = $scheme . '://' . $host . $port . $base_url . $path . $query . $fragment;
+
+    // If $url already has the full "real url", we don't have to modify it any further.
+    if(isset($GLOBALS['real_base_url']) && strpos($url, $GLOBALS['real_base_url']) !== 0) {
+      if (isset($GLOBALS['real_base_url']) && isset($GLOBALS['real_base_host'])) {
+        $url = str_replace($scheme . '://' . $host . $GLOBALS['real_base_path'], $GLOBALS['real_base_host'] . $GLOBALS['real_base_path'], $url);
+        $url = str_replace($scheme . '://' . $host, $GLOBALS['real_base_url'], $url);
+      }
+    }
     return $collect_bubbleable_metadata ? $generated_url->setGeneratedUrl($url) : $url;
   }
