--- core/modules/menu_ui/menu_ui.module	2018-11-07 19:36:14.000000000 +0000
+++ core/modules/menu_ui/menu_ui.module	2018-11-07 20:16:24.000000000 +0000
@@ -11,6 +11,7 @@
 use Drupal\Core\Breadcrumb\Breadcrumb;
 use Drupal\Core\Cache\CacheableMetadata;
 use Drupal\Core\Block\BlockPluginInterface;
+use Drupal\Core\Language\LanguageInterface;
 use Drupal\Core\Link;
 use Drupal\Core\Menu\MenuLinkInterface;
 use Drupal\Core\Form\FormStateInterface;
@@ -141,6 +142,7 @@
   /** @var \Drupal\node\NodeTypeInterface $node_type */
   $node_type = $node->type->entity;
   $menu_name = strtok($node_type->getThirdPartySetting('menu_ui', 'parent', 'main:'), ':');
+  $langcode = $node->language()->getId();
   $defaults = FALSE;
   if ($node->id()) {
     $id = FALSE;
@@ -150,6 +152,7 @@
       $query = \Drupal::entityQuery('menu_link_content')
         ->condition('link.uri', 'node/' . $node->id())
         ->condition('menu_name', $menu_name)
+        ->condition('langcode', array($langcode, LanguageInterface::LANGCODE_NOT_SPECIFIED, LanguageInterface::LANGCODE_NOT_APPLICABLE), 'IN')
         ->sort('id', 'ASC')
         ->range(0, 1);
       $result = $query->execute();
@@ -161,6 +164,7 @@
       $query = \Drupal::entityQuery('menu_link_content')
         ->condition('link.uri', 'entity:node/' . $node->id())
         ->condition('menu_name', array_values($type_menus), 'IN')
+        ->condition('langcode', array($langcode, LanguageInterface::LANGCODE_NOT_SPECIFIED, LanguageInterface::LANGCODE_NOT_APPLICABLE), 'IN')
         ->sort('id', 'ASC')
         ->range(0, 1);
       $result = $query->execute();
@@ -229,7 +233,10 @@
   $type_menus = Menu::loadMultiple($type_menus_ids);
   $available_menus = [];
   foreach ($type_menus as $menu) {
-    $available_menus[$menu->id()] = $menu->label();
+    // Only menus with the language of the current node, not specified, or not applicable will be used.
+    if (in_array($menu->language()->getId(), [$node->language()->getId(), LanguageInterface::LANGCODE_NOT_SPECIFIED, LanguageInterface::LANGCODE_NOT_APPLICABLE])) {
+      $available_menus[$menu->id()] = $menu->label();
+    }
   }
   if ($defaults['id']) {
     $default = $defaults['menu_name'] . ':' . $defaults['parent'];
