--- core/modules/system/src/Form/ThemeSettingsForm.php	2019-12-10 13:25:23.000000000 +0000
+++ core/modules/system/src/Form/ThemeSettingsForm.php	2019-12-10 13:30:54.000000000 +0000
@@ -464,9 +464,11 @@
     // If the user uploaded a new logo or favicon, save it to a permanent location
     // and use it in place of the default theme-provided file.
     $default_scheme = $this->config('system.file')->get('default_scheme');
+    // Can't use private for these files as they will not be accesible.
+    $file_default_scheme = ($default_scheme == 'private')? 'public' : $default_scheme;
     try {
       if (!empty($values['logo_upload'])) {
-        $filename = $this->fileSystem->copy($values['logo_upload']->getFileUri(), $default_scheme . '://');
+        $filename = $this->fileSystem->copy($values['logo_upload']->getFileUri(), $file_default_scheme . '://');
         $values['default_logo'] = 0;
         $values['logo_path'] = $filename;
       }
@@ -476,7 +478,7 @@
     }
     try {
       if (!empty($values['favicon_upload'])) {
-        $filename = $this->fileSystem->copy($values['favicon_upload']->getFileUri(), $default_scheme . '://');
+        $filename = $this->fileSystem->copy($values['favicon_upload']->getFileUri(), $file_default_scheme . '://');
         $values['default_favicon'] = 0;
         $values['favicon_path'] = $filename;
         $values['toggle_favicon'] = 1;
