--- core/modules/locale/tests/src/Functional/LocaleImportFunctionalTest.php	2025-06-25 16:53:47.873971500 +0000
+++ core/modules/locale/tests/src/Functional/LocaleImportFunctionalTest.php	2025-06-25 17:26:43.995963962 +0000
@@ -399,6 +399,34 @@
   }
 
   /**
+   * Tests imported PO files are listed on file views.
+   */
+  public function testPoFileImportAndFileListing() {
+    $this->container
+      ->get('module_installer')
+      ->install(['system', 'user', 'file', 'views']);
+
+    $this->drupalLogin($this->drupalCreateUser([
+      'access administration pages',
+      'access files overview',
+      'administer languages',
+      'translate interface',
+    ]));
+
+    // Import a dummy PO file.
+    $filename = $this->importPoFile($this->getPoFile(), [
+      'langcode' => 'fr',
+    ]);
+
+    // Ensure that an exception is not thrown on the admin file view as a result
+    // of the restrictions on .po files.
+    // @see \Drupal\locale\StreamWrapper\TranslationsStream::getExternalUrl()
+    $this->drupalGet('admin/content/files');
+    $this->assertSession()->statusCodeEquals(200);
+    $this->assertSession()->pageTextContains($filename);
+  }
+
+  /**
    * Helper function: import a standalone .po file in a given language.
    *
    * @param string $contents
@@ -407,8 +435,11 @@
    *   (optional) Additional options to pass to the translation import form.
    * @param array $encodings
    *   (optional) The encoding of the file.
+   *
+   * @return string
+   *   The name of the file uploaded.
    */
-  public function importPoFile($contents, array $options = [], array $encodings = []): void {
+  public function importPoFile($contents, array $options = [], array $encodings = []) {
     $file_system = \Drupal::service('file_system');
     $name = $file_system->tempnam('temporary://', "po_") . '.po';
     foreach ($encodings as $encoding) {
@@ -419,6 +450,7 @@
     $this->drupalGet('admin/config/regional/translate/import');
     $this->submitForm($options, 'Import');
     $file_system->unlink($name);
+    return str_replace('temporary://', '', $name);
   }
 
   /**
