--- core/modules/user/src/Form/UserPasswordForm.php	2018-03-27 11:47:10.240128098 +0000
+++ core/modules/user/src/Form/UserPasswordForm.php	2018-03-27 11:47:46.808061754 +0000
@@ -138,7 +138,7 @@
     $mail = _user_mail_notify('password_reset', $account, $langcode);
     if (!empty($mail)) {
       $this->logger('user')->notice('Password reset instructions mailed to %name at %email.', ['%name' => $account->getUsername(), '%email' => $account->getEmail()]);
-      drupal_set_message($this->t('Further instructions have been sent to your email address.'));
+      drupal_set_message($this->t('If the account exists, an email has been sent with further instructions to reset the password.'));
     }

     $form_state->setRedirect('user.page');