--- modules/simple_sitemap/modules/simple_sitemap_engines/src/Entity/SimpleSitemapEngine.php    2022-08-24 22:25:04
+++ modules/simple_sitemap/modules/simple_sitemap_engines/src/Entity/SimpleSitemapEngine.php 2023-01-10 15:32:17
@@ -112,6 +112,7 @@
   public static function loadSitemapSubmissionEngines(): array {
     $ids = \Drupal::entityQuery('simple_sitemap_engine')
       ->exists('url')
+      ->accessCheck()
       ->execute();

     return static::loadMultiple($ids);
@@ -125,6 +126,7 @@
   public static function loadIndexNowEngines(): array {
     $ids = \Drupal::entityQuery('simple_sitemap_engine')
       ->exists('index_now_url')
+      ->accessCheck()
       ->execute();

     return static::loadMultiple($ids);
@@ -139,6 +141,7 @@
   public static function loadRandomIndexNowEngine(): ?SimpleSitemapEngine {
     if ($ids = \Drupal::entityQuery('simple_sitemap_engine')
       ->exists('index_now_url')
+      ->accessCheck()
       ->execute()) {
       return static::load(array_rand($ids));
     }

--- modules/simple_sitemap/modules/simple_sitemap_views/simple_sitemap_views.libraries.yml     2023-02-10 10:50:42
+++ modules/simple_sitemap/modules/simple_sitemap_views/simple_sitemap_views.libraries.yml        2023-02-10 10:50:38
@@ -2,6 +2,7 @@
   version: VERSION
   js:
     js/simple_sitemap.viewsUi.js: {}
+    js/simple_sitemap.sitemapEntities.js: {}
   dependencies:
     - core/jquery
     - core/drupal
