--- modules/clipboardjs/templates/clipboardjs-button.html.twig  2022-04-07 10:53:19
+++ modules/clipboardjs/templates/clipboardjs-button.html.twig     2023-01-09 11:53:51
@@ -19,7 +19,7 @@
  */
 #}
 {{ attach_library('clipboardjs/button') }}
-{% spaceless %}
+{% apply spaceless %}
   <span class="clipboardjs clipboardjs-btn">
     <button class="clipboardjs-button" data-clipboard-alert="{{ alert_style }}" data-clipboard-alert-text="{{ alert_text }}" data-clipboard-text="{{ value }}">
       <div class="tooltip">
@@ -28,4 +28,4 @@
       </div>
     </button>
   </span>
-{% endspaceless %}
+{% endapply %}

--- modules/clipboardjs/templates/clipboardjs-snippet.html.twig    2023-01-09 11:55:54
+++ modules/clipboardjs/templates/clipboardjs-snippet.html.twig 2023-01-09 11:56:05
@@ -19,7 +19,7 @@
  */
 #}
 {{ attach_library('clipboardjs/snippet') }}
-{% spaceless %}
+{% apply spaceless %}
   <pre class="clipboardjs clipboardjs-snippet">
     <button class="clipboardjs-button" data-clipboard-alert="{{ alert_style }}" data-clipboard-alert-text="{{ alert_text }}" data-clipboard-target="#{{ attributes.id }}">
       <div class="tooltip">
@@ -29,4 +29,4 @@
     </button>
     <code id="{{ attributes.id }}"{{ attributes }}>{{ value }}</code>
   </pre>
-{% endspaceless %}
+{% endapply %}

--- modules/clipboardjs/templates/clipboardjs-textarea.html.twig        2022-04-07 10:53:19
+++ modules/clipboardjs/templates/clipboardjs-textarea.html.twig   2023-01-09 11:57:00
@@ -19,7 +19,7 @@
  */
 #}
 {{ attach_library('clipboardjs/textarea') }}
-{% spaceless %}
+{% apply spaceless %}
   <div class="clipboardjs clipboardjs-textarea">
     <textarea id="{{ attributes.id }}"{{ attributes }}>{{ value }}</textarea>
     <button class="clipboardjs-button" data-clipboard-alert="{{ alert_style }}" data-clipboard-alert-text="{{ alert_text }}" data-clipboard-target="#{{ attributes.id }}">
@@ -29,4 +29,4 @@
       </div>
     </button>
   </div>
-{% endspaceless %}
+{% endapply %}

--- modules/clipboardjs/templates/clipboardjs-textfield.html.twig       2022-04-07 10:53:19
+++ modules/clipboardjs/templates/clipboardjs-textfield.html.twig  2023-01-09 11:57:48
@@ -19,7 +19,7 @@
  */
 #}
 {{ attach_library('clipboardjs/textfield') }}
-{% spaceless %}
+{% apply spaceless %}
   <div class="clipboardjs clipboardjs-textfield">
     <input id="{{ attributes.id }}"{{ attributes.id }} type="text" class="clipboardjs" value="{{ value }}" readonly="readonly" />
     <button class="clipboardjs-button" data-clipboard-alert="{{ alert_style }}" data-clipboard-alert-text="{{ alert_text }}" data-clipboard-target="#{{ attributes.id }}">
@@ -29,4 +29,4 @@
       </div>
     </button>
   </div>
-{% endspaceless %}
+{% endapply %}

--- modules/clipboardjs/templates/clipboardjs.html.twig 2022-04-07 10:53:19
+++ modules/clipboardjs/templates/clipboardjs.html.twig    2023-01-09 11:58:38
@@ -19,7 +19,7 @@
  */
 #}
 {{ attach_library('clipboardjs/textfield') }}
-{% spaceless %}
+{% apply spaceless %}
   <input id="{{ attributes.id }}" type="text" class="clipboardjs" value="{{ value }}" readonly="readonly" />
   <a class="clipboardjs-button" data-clipboard-alert="{{ alert_style }}" data-clipboard-alert-text="{{ alert_text }}" data-clipboard-target="#{{ attributes.id }}">
     <div class="tooltip">
@@ -27,4 +27,4 @@
       <span class="tooltiptext"></span>
     </div>
   </a>
-{% endspaceless %}
+{% endapply %}
