--- modules/session_based_temp_store/src/SessionBasedTempStore.php      2023-01-05 16:47:31
+++ modules/session_based_temp_store/src/SessionBasedTempStore.php 2023-01-05 16:47:27
@@ -186,7 +186,7 @@
     $value = (object) [
       'owner' => $this->getOwner(),
       'data' => $value,
-      'updated' => (int) $this->requestStack->getMasterRequest()->server->get('REQUEST_TIME'),
+      'updated' => (int) $this->requestStack->getMainRequest()->server->get('REQUEST_TIME'),
     ];

     // If the global expiration time is set to 0 (expire at the end of the session),
@@ -353,7 +353,7 @@
       return $this->expire;
     }
     // Otherwise set the specific expiration time passed in the argument.
-    $request_time = (int) $this->requestStack->getMasterRequest()->server->get('REQUEST_TIME');
+    $request_time = (int) $this->requestStack->getMainRequest()->server->get('REQUEST_TIME');
     return $request_time + $this->expire;
   }
