--- modules/graphql_twig/graphql_twig.info.yml  2022-12-15 10:08:37
+++ modules/graphql_twig/graphql_twig.info.yml     2022-12-15 15:21:39
@@ -2,8 +2,7 @@
 type: module
 description: 'Render twig templates with graphql data.'
 package: GraphQL
-core: 8.x
-core_version_requirement: ^8 || ^9
+core_version_requirement: ^8 || ^9 || ^10
 dependencies:
   - graphql:graphql (< 4.0)
   - graphql:graphql_core

diff --git  modules/graphql_twig/tests/src/Kernel/BlockTest.php  modules/graphql_twig/tests/src/Kernel/BlockTest.php
index c31853e..0c9912e 100644
--- modules/graphql_twig/tests/src/Kernel/BlockTest.php
+++ modules/graphql_twig/tests/src/Kernel/BlockTest.php
@@ -17,7 +17,7 @@ class BlockTest extends GraphQLTestBase {
   /**
    * {@inheritdoc}
    */
-  public static $modules = [
+  protected static $modules = [
     'block',
     'graphql_twig',
   ];
diff --git  modules/graphql_twig/tests/src/Kernel/EntityRenderTest.php  modules/graphql_twig/tests/src/Kernel/EntityRenderTest.php
index e263fe9..e804f03 100644
--- modules/graphql_twig/tests/src/Kernel/EntityRenderTest.php
+++ modules/graphql_twig/tests/src/Kernel/EntityRenderTest.php
@@ -22,7 +22,7 @@ class EntityRenderTest extends KernelTestBase {
   /**
    * {@inheritdoc}
    */
-  public static $modules = [
+  protected static $modules = [
     'system',
     'graphql',
     'graphql_core',
diff --git  modules/graphql_twig/tests/src/Kernel/RouteTest.php  modules/graphql_twig/tests/src/Kernel/RouteTest.php
index c77176b..a567229 100644
--- modules/graphql_twig/tests/src/Kernel/RouteTest.php
+++ modules/graphql_twig/tests/src/Kernel/RouteTest.php
@@ -15,7 +15,7 @@ class RouteTest extends GraphQLTestBase {
   /**
    * {@inheritdoc}
    */
-  public static $modules = [
+  protected static $modules = [
     'graphql_twig',
   ];

diff --git  modules/graphql_twig/tests/src/Kernel/ThemeTest.php  modules/graphql_twig/tests/src/Kernel/ThemeTest.php
index 838c59a..0cc5ba8 100644
--- modules/graphql_twig/tests/src/Kernel/ThemeTest.php
+++ modules/graphql_twig/tests/src/Kernel/ThemeTest.php
@@ -27,7 +27,7 @@ class ThemeTest extends KernelTestBase {
   /**
    * {@inheritdoc}
    */
-  public static $modules = [
+  protected static $modules = [
     'system',
     'graphql',
     'graphql_twig',

--- modules/graphql_twig/graphql_twig.libraries.yml     2023-01-13 12:03:14
+++ modules/graphql_twig/graphql_twig.libraries.yml        2023-01-13 12:03:11
@@ -7,5 +7,5 @@
     assets/debug.js: {}
   dependencies:
     - core/jquery
-    - core/jquery.once
+    - core/once
     - core/drupal

--- modules/graphql_twig/assets/debug.js        2023-01-13 12:05:16
+++ modules/graphql_twig/assets/debug.js   2023-01-13 12:05:20
@@ -1,7 +1,7 @@
-(function ($, Drupal) {
+(function ($, Drupal, once) {
   Drupal.behaviors.graphqlTwigDebug = {
     attach: function (context, settings) {
-      $('.graphql-twig-debug-wrapper', context).once('graphql-debug').each(function () {
+      $(once('graphql-debug', '.graphql-twig-debug-wrapper', context).each(function () {
         var query = $(this).attr('data-graphql-query'),
             variables = $(this).attr('data-graphql-variables'),
             $form = $('<form method="post" target="_blank"></form>').attr('action', Drupal.url('graphql/explorer')).appendTo(this),
@@ -15,4 +15,4 @@
       });
     }
   };
-}(jQuery, Drupal));
+}(jQuery, Drupal, once));
