--- modules/captcha/image_captcha/src/Response/CaptchaImageResponse.php 2023-01-09 12:34:09
+++ modules/captcha/image_captcha/src/Response/CaptchaImageResponse.php    2023-01-09 12:34:03
@@ -56,7 +56,7 @@
   /**
    * {@inheritdoc}
    */
-  public function __construct(Config $config, LoggerInterface $logger, Connection $connection, FileSystemInterface $fileSystem, $callback = NULL, $status = 200, $headers = []) {
+  public function __construct(Config $config, LoggerInterface $logger, Connection $connection, FileSystemInterface $fileSystem, $status = 200, $headers = []) {
     parent::__construct(NULL, $status, $headers);

     $this->config = $config;
@@ -68,7 +68,7 @@
   /**
    * {@inheritdoc}
    */
-  public function prepare(Request $request) {
+  public function prepare(Request $request): static {
     $session_id = $request->get('session_id');

     $code = $this->connection
@@ -92,7 +92,7 @@
   /**
    * {@inheritdoc}
    */
-  public function sendHeaders() {
+  public function sendHeaders(): static {
     if ($this->config->get('image_captcha_file_format') == IMAGE_CAPTCHA_FILE_FORMAT_JPG) {
       $this->headers->set('content-type', 'image/jpeg');
     }
@@ -107,9 +107,9 @@
   /**
    * {@inheritdoc}
    */
-  public function sendContent() {
+  public function sendContent(): static {
     if (!$this->image) {
-      return;
+      return $this;
     }

     // Begin capturing the byte stream.
@@ -122,8 +122,7 @@
     else {
       imagepng($this->image);
     }
-    // Clean up the image resource.
-    imagedestroy($this->image);
+    return $this;
   }

   /**

--- modules/captcha/image_captcha/src/StreamedResponse/CaptchaFontPreviewStreamedResponse.php   2023-01-09 12:38:57
+++ modules/captcha/image_captcha/src/StreamedResponse/CaptchaFontPreviewStreamedResponse.php      2023-01-09 12:37:47
@@ -27,7 +27,7 @@
   /**
    * {@inheritdoc}
    */
-  public function __construct(ImmutableConfig $config, $token, $callback = NULL, $status = 200, $headers = []) {
+  public function __construct(ImmutableConfig $config, $token, $status = 200, $headers = []) {
     parent::__construct(NULL, $status, $headers);

     $this->config = $config;
@@ -37,7 +37,7 @@
   /**
    * {@inheritdoc}
    */
-  public function sendContent() {
+  public function sendContent(): static {
     // Get the font from the given font token.
     if ($this->token == 'BUILTIN') {
       $font = 'BUILTIN';
@@ -46,14 +46,15 @@
       // Get the mapping of font tokens to font file objects.
       $fonts = $this->config->get('image_captcha_fonts_preview_map_cache');
       if (!isset($fonts[$this->token])) {
-        return 'bad token';
+        throw new \LogicException('bad token');
       }
       // Get the font path.
       $font = $fonts[$this->token]['uri'];
       // Some sanity checks if the given font is valid.
       if (!is_file($font) || !is_readable($font)) {
-        return 'bad font';
+        throw new \LogicException('bad font');
       }
+      return $this;
     }

     // Settings of the font preview.
@@ -83,8 +84,7 @@
     $this->headers->set('Content-Type', 'image/png');
     // Dump image data to client.
     imagepng($image);
-    // Release image memory.
-    imagedestroy($image);
+    return $this;
   }

 }

--- modules/captcha/src/Form/CaptchaPointForm.php       2023-01-09 12:41:29
+++ modules/captcha/src/Form/CaptchaPointForm.php  2023-01-09 12:41:25
@@ -113,11 +113,15 @@
       $this->messenger()->addMessage($this->t('Captcha Point for %form_id form was created.', [
         '%form_id' => $captcha_point->getFormId(),
       ]));
+      $form_state->setRedirect('captcha_point.list');
+      return SAVED_NEW;
     }
     else {
       $this->messenger()->addMessage($this->t('Captcha Point for %form_id form was updated.', [
         '%form_id' => $captcha_point->getFormId(),
       ]));
+      $form_state->setRedirect('captcha_point.list');
+      return SAVED_UPDATED;
     }
     $form_state->setRedirect('captcha_point.list');
   }

--- modules/captcha/captcha.install     2023-01-10 13:05:14
+++ modules/captcha/captcha.install        2023-01-10 13:05:09
@@ -153,7 +153,7 @@
 function captcha_update_8902(&$sandbox) {
   $query = \Drupal::entityQuery('captcha_point');
   $query->notExists('label');
-  $entity_ids = $query->execute();
+  $entity_ids = $query->accessCheck()->execute();

 if (!empty($entity_ids) && is_array($entity_ids)) {
   foreach($entity_ids as $entity_id) {

--- modules/captcha/captcha.module 2022-12-31 11:04:08
+++ modules/captcha/captcha.module      2023-01-10 13:06:28
@@ -173,13 +173,13 @@
   if (!$account->hasPermission('skip CAPTCHA')) {
     $query = \Drupal::entityQuery('captcha_point');
     $query->condition('label', $form_id);
-    $entity_ids = $query->execute();
+    $entity_ids = $query->accessCheck()->execute();

     // If empty, see if it is a form provided by default config.
     if (empty($entity_ids)) {
       $query = \Drupal::entityQuery('captcha_point');
       $query->condition('formId', $form_id);
-      $entity_ids = $query->execute();
+      $entity_ids = $query->accessCheck()->execute();
     }

     if (!empty($entity_ids) && is_array($entity_ids)) {

--- modules/captcha/image_captcha/image_captcha.libraries.yml   2022-12-31 11:04:08
+++ modules/captcha/image_captcha/image_captcha.libraries.yml      2023-01-13 11:58:37
@@ -9,7 +9,6 @@
     - core/jquery
     - core/drupal
     - core/drupalSettings
-    - core/jquery.once

 image-captcha-refresh:
   version: 1.0
