diff --git modules/securitytxt/securitytxt.info.yml modules/securitytxt/securitytxt.info.yml
index 1e2f17f..6dfe4d2 100644
--- modules/securitytxt/securitytxt.info.yml
+++ modules/securitytxt/securitytxt.info.yml
@@ -2,6 +2,5 @@ name: Security.txt
 type: module
 description: 'Provides a security.txt file for your site.'
 package: Security
-core: '8.x'
-core_version_requirement: ^8 || ^9
+core_version_requirement: ^9.1 || ^10
 configure: securitytxt.configure
diff --git modules/securitytxt/tests/src/Functional/SecuritytxtBaseTest.php modules/securitytxt/tests/src/Functional/SecuritytxtBaseTest.php
index 695909a..f5342f4 100644
--- modules/securitytxt/tests/src/Functional/SecuritytxtBaseTest.php
+++ modules/securitytxt/tests/src/Functional/SecuritytxtBaseTest.php
@@ -17,7 +17,7 @@ abstract class SecuritytxtBaseTest extends BrowserTestBase {
    * @var array
    *  Modules which should be enabled by default.
    */
-  public static $modules = ['securitytxt'];
+  protected static $modules = ['securitytxt'];

   /**
    * User with no permissions.
@@ -50,7 +50,7 @@ abstract class SecuritytxtBaseTest extends BrowserTestBase {
   /**
    * {@inheritdoc}
    */
-  protected function setUp() {
+  protected function setUp(): void {
     parent::setUp();

     $this->authenticatedUser = $this->drupalCreateUser([]);
@@ -91,7 +91,8 @@ abstract class SecuritytxtBaseTest extends BrowserTestBase {
     $path = 'admin/config/system/securitytxt';
     $submit = 'Save configuration';
     $options = [];
-    $this->drupalPostForm($path, $edit, $submit, $options);
+    $this->drupalGet($path, $options);
+    $this->submitForm($edit, $submit);
   }

   /**
@@ -105,7 +106,8 @@ abstract class SecuritytxtBaseTest extends BrowserTestBase {
     $path = 'admin/config/system/securitytxt/sign';
     $submit = 'Save configuration';
     $options = [];
-    $this->drupalPostForm($path, $edit, $submit, $options);
+    $this->drupalGet($path, $options);
+    $this->submitForm($edit, $submit);
   }

   /**
