--- modules/taxonomy_manager/src/Element/TaxonomyManagerTree.php        2023-01-10 15:36:34
+++ modules/taxonomy_manager/src/Element/TaxonomyManagerTree.php   2023-01-10 15:36:27
@@ -95,7 +95,7 @@
       if ($pager_size > 0) {
         $query->pager($pager_size);
       }
-      $tids = $query->execute();
+      $tids = $query->accessCheck()->execute();
       return \Drupal::entityTypeManager()->getStorage('taxonomy_term')->loadMultiple($tids);
     }
     catch (QueryException $e) {

--- modules/taxonomy_manager/taxonomy_manager.libraries.yml     2023-01-13 11:31:13
+++ modules/taxonomy_manager/taxonomy_manager.libraries.yml        2023-01-13 12:18:07
@@ -25,7 +25,7 @@
   dependencies:
     - taxonomy_manager/jquery.fancytree
     - core/drupalSettings
-    - core/jquery.once
+    - core/once

 termData:
   js:

--- modules/taxonomy_manager/js/tree.js 2023-01-13 12:22:02
+++ modules/taxonomy_manager/js/tree.js    2023-01-13 12:21:51
@@ -1,4 +1,4 @@
-(function ($, Drupal, drupalSettings) {
+(function ($, Drupal, drupalSettings, once) {
   'use strict';

   /**
@@ -9,7 +9,7 @@
       var treeSettings = settings.taxonomy_manager.tree || [];
       if (treeSettings instanceof Array) {
         for (var i = 0; i < treeSettings.length; i++) {
-          $('#' + treeSettings[i].id).once('taxonomy-manager-tree').each(function () {
+          $(once('taxonomy-manager-tree', '#' + treeSettings[i].id)).each(function () {
             var tree = new Drupal.TaxonomyManagerFancyTree(treeSettings[i].id, treeSettings[i].name, treeSettings[i].source);
           });
         }
@@ -21,7 +21,7 @@
         return true;
       });
       // Handle click on autocomplete suggestion
-      $('.ui-autocomplete', context).once('input').on('click', function (e) {
+      $(once('input', '.ui-autocomplete', context)).on('click', function (e) {
         e.stopPropagation();
         var tidMatch = $('input[name="search_terms"]').val().match(/\([0-9]*\)/g)
         if (tidMatch.length) {
@@ -108,4 +108,4 @@
     });
   };

-})(jQuery, Drupal, drupalSettings);
+})(jQuery, Drupal, drupalSettings, once);
