--- modules/warmer/modules/warmer_entity/src/Plugin/warmer/EntityWarmer.php     2022-12-20 15:40:05
+++ modules/warmer/modules/warmer_entity/src/Plugin/warmer/EntityWarmer.php        2022-12-20 15:40:00
@@ -140,7 +140,7 @@
         if (!empty($bundle_key)) {
           $query->condition($bundle_key, $bundle);
         }
-        $results = $query->execute();
+        $results = $query->accessCheck()->execute();
         $entity_ids = array_filter((array) array_values($results));
         $iids = array_merge($iids, array_map(
           function ($id) use ($entity_type_id) {
