--- modules/captcha/captcha.install     2023-01-10 13:05:14
+++ modules/captcha/captcha.install        2023-01-10 13:05:09
@@ -153,7 +153,7 @@
 function captcha_update_8902(&$sandbox) {
   $query = \Drupal::entityQuery('captcha_point');
   $query->notExists('label');
-  $entity_ids = $query->execute();
+  $entity_ids = $query->accessCheck()->execute();

 if (!empty($entity_ids) && is_array($entity_ids)) {
   foreach($entity_ids as $entity_id) {

--- modules/captcha/captcha.module 2022-12-31 11:04:08
+++ modules/captcha/captcha.module      2023-01-10 13:06:28
@@ -173,13 +173,13 @@
   if (!$account->hasPermission('skip CAPTCHA')) {
     $query = \Drupal::entityQuery('captcha_point');
     $query->condition('label', $form_id);
-    $entity_ids = $query->execute();
+    $entity_ids = $query->accessCheck()->execute();

     // If empty, see if it is a form provided by default config.
     if (empty($entity_ids)) {
       $query = \Drupal::entityQuery('captcha_point');
       $query->condition('formId', $form_id);
-      $entity_ids = $query->execute();
+      $entity_ids = $query->accessCheck()->execute();
     }

     if (!empty($entity_ids) && is_array($entity_ids)) {
