--- modules/flag/src/FlagService.php    2023-01-10 14:40:56
+++ modules/flag/src/FlagService.php       2023-01-10 14:40:58
@@ -74,7 +74,7 @@
       $query->condition('entity_type', $entity_type);
     }

-    $ids = $query->execute();
+    $ids = $query->accessCheck()->execute();
     $flags = $this->getFlagsByIds($ids);

     if (isset($bundle)) {
@@ -200,7 +200,7 @@
     $query->condition('entity_type', $entity->getEntityTypeId())
       ->condition('entity_id', $entity->id());

-    $ids = $query->execute();
+    $ids = $query->accessCheck()->execute();

     return $this->getFlaggingsByIds($ids);
   }
@@ -230,7 +230,7 @@
     $query->condition('entity_type', $entity->getEntityTypeId())
       ->condition('entity_id', $entity->id());

-    $ids = $query->execute();
+    $ids = $query->accessCheck()->execute();

     return $this->getFlaggingsByIds($ids);
   }
@@ -263,7 +263,7 @@
       $query->condition('flag_id', $flag->id());
     }

-    $ids = $query->execute();
+    $ids = $query->accessCheck()->execute();
     // Load the flaggings.
     $flaggings = $this->getFlaggingsByIds($ids);

@@ -352,7 +352,7 @@
     $query->accessCheck();
     $query->condition('flag_id', $flag->id());

-    $ids = $query->execute();
+    $ids = $query->accessCheck()->execute();

     $flaggings = $this->getFlaggingsByIds($ids);

@@ -369,7 +369,7 @@
       ->condition('entity_type', $entity->getEntityTypeId())
       ->condition('entity_id', $entity->id());

-    $ids = $query->execute();
+    $ids = $query->accessCheck()->execute();

     $flaggings = $this->getFlaggingsByIds($ids);

@@ -392,7 +392,7 @@
       $query->condition('session_id', $session_id);
     }

-    $ids = $query->execute();
+    $ids = $query->accessCheck()->execute();

     $flaggings = $this->getFlaggingsByIds($ids);
