--- modules/flood_unblock/src/Commands/FloodUnblockCommands.php	2020-03-19 07:47:09.000000000 +0000
+++ modules/flood_unblock/src/Commands/FloodUnblockCommands.php	2020-07-30 10:31:38.000000000 +0100
@@ -38,6 +38,7 @@
    */
   public function unblockIp($ip = NULL) {
     $this->manager->floodUnblockClearEvent('user.failed_login_ip', $ip);
+    $this->manager->floodUnblockClearEvent('user.password_request_ip', NULL);

     $this->output()->writeln('Done');
   }
@@ -54,6 +55,7 @@
   public function unblockUser($user = NULL) {
     $this->manager->floodUnblockClearEvent('user.failed_login_user', $user);
     $this->manager->floodUnblockClearEvent('user.http_login', $user);
+    $this->manager->floodUnblockClearEvent('user.password_request_user', NULL);
     $this->output()->writeln('Done');
   }

@@ -67,6 +69,8 @@
     $this->manager->floodUnblockClearEvent('user.failed_login_ip', NULL);
     $this->manager->floodUnblockClearEvent('user.failed_login_user', NULL);
     $this->manager->floodUnblockClearEvent('user.http_login', NULL);
+    $this->manager->floodUnblockClearEvent('user.password_request_ip', NULL);
+    $this->manager->floodUnblockClearEvent('user.password_request_user', NULL);
     $this->output()->writeln('Done');
   }
