--- modules/graphql_twig/tests/src/Traits/ThemeTestTrait.php	2019-10-30 14:26:29.000000000 +0000
+++ modules/graphql_twig/tests/src/Traits/ThemeTestTrait.php	2019-10-30 14:28:11.000000000 +0000
@@ -2,7 +2,7 @@
 
 namespace Drupal\Tests\graphql_twig\Traits;
 
-use Drupal\Core\Session\AccountProxy;
+use Drupal\Core\Session\AccountProxyInterface;
 use Drupal\graphql\GraphQL\Execution\QueryProcessor;
 use Drupal\KernelTests\KernelTestBase;
 use Prophecy\Argument;
@@ -21,7 +21,7 @@
   protected function setupThemeTest() {
     if ($this instanceof KernelTestBase) {
       // Mock a user that is allowed to do everything.
-      $currentUser = $this->prophesize(AccountProxy::class);
+      $currentUser = $this->prophesize(AccountProxyInterface::class);
       $currentUser->isAuthenticated()->willReturn(TRUE);
       $currentUser->hasPermission(Argument::any())->willReturn(TRUE);
       $currentUser->id()->willReturn(1);
