--- modules/password_policy/src/PasswordPolicyValidator.php
+++ modules/password_policy/src/PasswordPolicyValidator.php
@@ -61,6 +61,9 @@
     $original_roles = array_combine($original_roles, $original_roles);
 
     $force_failure = FALSE;
+    if (is_array($password) && array_key_exists('pass1', $password)) {
+      $password = $password['pass1'];
+    }
     if ($edited_user_roles !== $original_roles && $password === '' && !empty($applicable_policies)) {
       // New role has been added and applicable policies are available.
       $force_failure = TRUE;