--- modules/content_synchronizer/src/Processors/ImportProcessor.php     2023-07-03 16:41:58
+++ modules/content_synchronizer/src/Processors/ImportProcessor.php        2023-08-22 12:19:12
@@ -19,7 +19,7 @@
   const UPDATE_SYSTEMATIC = 'update_systematic';
   const UPDATE_IF_RECENT = 'update_if_recent';
   const UPDATE_NO_UPDATE = 'update_no_update';
-  const DEFAULT_UPDATE_TYPE = 'update_if_recent';
+  const DEFAULT_UPDATE_TYPE = 'update_systematic';

   /**
    * The current import processor.
