--- modules/social_post/src/SocialPostManager.php	2019-10-30 14:36:24.000000000 +0000
+++ modules/social_post/src/SocialPostManager.php	2019-10-30 14:38:08.000000000 +0000
@@ -5,7 +5,7 @@
 use Drupal\Core\Entity\EntityStorageException;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Logger\LoggerChannelFactoryInterface;
-use Drupal\Core\Session\AccountProxy;
+use Drupal\Core\Session\AccountProxyInterface;
 use Drupal\Core\StringTranslation\StringTranslationTrait;
 
 /**
@@ -25,7 +25,7 @@
   /**
    * The current logged in Drupal user.
    *
-   * @var \Drupal\Core\Session\AccountProxy
+   * @var \Drupal\Core\Session\AccountProxyInterface
    */
   protected $currentUser;
 
@@ -62,7 +62,7 @@
    *
    * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
    *   Used for loading and creating Drupal user objects.
-   * @param \Drupal\Core\Session\AccountProxy $current_user
+   * @param \Drupal\Core\Session\AccountProxyInterface $current_user
    *   Used to get current active user.
    * @param \Drupal\social_post\SocialPostDataHandler $data_handler
    *   Used to handle session values.
@@ -70,7 +70,7 @@
    *   Used for logging errors.
    */
   public function __construct(EntityTypeManagerInterface $entity_type_manager,
-                              AccountProxy $current_user,
+                              AccountProxyInterface $current_user,
                               SocialPostDataHandler $data_handler,
                               LoggerChannelFactoryInterface $logger_factory) {
 
