--- modules/clipboardjs/js/clipboard.js	2021-10-12 08:56:03.000000000 +0100
+++ modules/clipboardjs/js/clipboard.js	2021-10-12 08:56:43.000000000 +0100
@@ -20,7 +20,10 @@
         Drupal.clipboard.on('success', function (e) {
           let alertStyle = $(e.trigger).data('clipboardAlert');
           let alertText = $(e.trigger).data('clipboardAlertText');
-          let target = $(e.trigger).data('clipboardTarget');
+          let target = $(e.trigger.parentNode).find('input');
+          if (!$(target.length)) {
+            target = $(e.trigger).data('clipboardTarget');
+          }
 
           // Display as alert.
           if (alertStyle === 'alert') {
