--- modules/field_group/field_group.module	2019-06-10 15:14:13.000000000 +0100
+++ modules/field_group/field_group.module	2019-06-10 15:14:53.000000000 +0100
@@ -315,7 +315,7 @@
       }
 
       $group_parents = $element['#array_parents'];
-      $group_parents[] = empty($group->parent_name) ? $group->region : $group->parent_name;
+      $group_parents[] = !(empty($group->parent_name)) ? $group->parent_name : $group->region;
       $group_references[$group_name] = &$element[$group_name];
       $element[$group_name]['#group'] = implode('][', $group_parents);
 
