--- modules/password_policy/password_policy_username/src/Plugin/PasswordConstraint/PasswordUsername.php
+++ modules/password_policy/password_policy_username/src/Plugin/PasswordConstraint/PasswordUsername.php
@@ -25,8 +25,8 @@ class PasswordUsername extends PasswordConstraintBase {
   public function validate($password, UserInterface $user) {
     $config = $this->getConfiguration();
     $validation = new PasswordPolicyValidation();
-
-    if ($config['disallow_username'] && stripos($password, $user->getAccountName()) !== FALSE) {
+    $username = $user->getAccountName();
+    if ($config['disallow_username'] && $username != "" && stripos($password, $username) !== FALSE) {
       $validation->setErrorMessage($this->t('Password must not contain the username.'));
     }
 
