--- modules/password_policy/src/PasswordPolicyValidator.php	2022-09-28 15:02:47.000000000 +0100
+++ modules/password_policy/src/PasswordPolicyValidator.php	2022-09-28 15:03:16.000000000 +0100
@@ -79,6 +79,9 @@
     $original_roles = array_combine($original_roles, $original_roles);

     $force_failure = FALSE;
+    if (is_array($password) && array_key_exists('pass1', $password)) {
+      $password = $password['pass1'];
+    }
     if (!empty(array_diff($edited_user_roles, $original_roles)) && $password === '' && !empty($applicable_policies)) {
       // New role has been added and applicable policies are available.
       $force_failure = TRUE;
