--- modules/redirect/src/EventSubscriber/RouteNormalizerRequestSubscriber.php
+++ modules/redirect/src/EventSubscriber/RouteNormalizerRequestSubscriber.php
@@ -123,7 +123,12 @@

       $original_uri = $request->getSchemeAndHttpHost() . $request->getRequestUri();
       $original_uri = urldecode($original_uri);
-      if (urldecode($redirect_uri) != $original_uri) {
+      $original_uri_parsed = parse_url($original_uri);
+      $redirect_uri_parsed = parse_url(urldecode($redirect_uri));
+      if ($original_uri_parsed['path'] !== $redirect_uri_parsed['path']) {
+        if ($original_uri_parsed['host'] !== $redirect_uri_parsed['host']){
+          $redirect_uri = str_replace($redirect_uri_parsed['host'], $original_uri_parsed['host'], $redirect_uri);
+        }
         $response = new TrustedRedirectResponse($redirect_uri, $this->config->get('default_status_code'));
         $response->headers->set('X-Drupal-Route-Normalizer', 1);
         $event->setResponse($response);
