--- modules/view_password/js/password.js        2025-05-06 14:24:23.413899728 +0000
+++ modules/view_password/js/password.js        2025-05-06 14:24:11.924899772 +0000
@@ -16,7 +16,7 @@
       $(once('view_password_button', '.pwd-see [type=password]', context))
         .after(
           `<button type="button" class="shwpd ${span_classes_custom} eye-close"
-            aria-label="${drupalSettings.view_password.showPasswordLabel}"
+            aria-label="${drupalSettings.view_password.showPasswordLabel}" data-toggle="tooltip" data-placement="auto" title=${Drupal.t('Show')}
           >
           </button>`
         );
@@ -36,7 +36,18 @@
               .siblings(':password')
               .prop('type', 'text');
             $(this)
-              .attr('aria-label', drupalSettings.view_password.hidePasswordLabel);
+              .attr({
+                'aria-label': 'drupalSettings.view_password.hidePasswordLabel',
+                'data-toggle': 'tooltip',
+                'data-placement': 'auto',
+                'title': `${Drupal.t('Hide')}`
+              });
+            // autohide after 20s
+            setTimeout(function() {
+              $(this).addClass("eye-close").removeClass("eye-open");
+              $('.eye-close', context).prev(':text').prop('type', 'password');
+              $('button.shwpd').attr('aria-label', drupalSettings.view_password.showPasswordLabel);
+            }, 20000);
             if (icon_exposed_custom !== '') {
               $(this).css({'background-image': `url(${icon_exposed_custom})`});
             }
@@ -46,7 +57,12 @@
               .siblings(':text')
               .prop('type', 'password');
             $(this)
-              .attr('aria-label', drupalSettings.view_password.showPasswordLabel);
+              .attr({
+                'aria-label': 'drupalSettings.view_password.showPasswordLabel',
+                'data-toggle': 'tooltip',
+                'data-placement': 'auto',
+                'title': `${Drupal.t('Show')}`
+              });
             if (icon_hidden_custom !== '') {
               $(this).css({'background-image': `url(${icon_hidden_custom})`});
             }