--- modules/view_password/js/password.js	2023-01-27 19:04:57.475244610 +0000
+++ modules/view_password/js/password.js	2023-01-27 19:21:09.970208807 +0000
@@ -24,6 +24,12 @@
               .siblings(':password')
               .prop('type', 'text');
             $('button.shwpd').attr('aria-label', drupalSettings.view_password.hidePasswordLabel);
+            // autohide after 20s
+            setTimeout(function() {
+              $('.shwpd').addClass("eye-close").removeClass("eye-open");
+              $('.eye-close', context).prev(':text').prop('type', 'password');
+              $('button.shwpd').attr('aria-label', drupalSettings.view_password.showPasswordLabel);
+            }, 20000);
           } else if ($(this).hasClass('eye-close')) {
             $('.eye-close', context)
               .siblings(':text')