--- modules/paranoia/paranoia.module	2018-02-11 13:02:55.000000000 +0000
+++ modules/paranoia/paranoia.module	2018-02-11 13:04:19.000000000 +0000
@@ -34,7 +34,7 @@
  */
 function paranoia_user_role_presave(RoleInterface $role) {
   $role->setIsAdmin(FALSE);
-  \Drupal::service('logger.channel.paranoia')->alert('An attempt to elevate the role %role_label to admin was detected and blocked.', ['%role_label' => $role->label()]);
+  \Drupal::service('logger.channel.paranoia')->info('An attempt to elevate the role %role_label to admin was detected and blocked.', ['%role_label' => $role->label()]);
 }
 
 /**
