--- modules/single_content_sync/src/ContentImporter.php 2023-05-10 10:26:43
+++ modules/single_content_sync/src/ContentImporter.php    2023-05-10 10:26:30
@@ -558,6 +558,8 @@
    * {@inheritdoc}
    */
   public function mapBaseFieldsValues(string $entity_type_id, array $values): array {
+    $entity = [];
+
     switch ($entity_type_id) {
       case 'node':
         $entity = [
@@ -620,9 +622,6 @@
           'status' => $values['status'],
         ];
         break;
-
-      default:
-        return [];
     }

     // Set moderation state if it is supported for multiple entities.
@@ -630,6 +629,10 @@
       $entity['moderation_state'] = $values['moderation_state'];
     }

+    // Alter entity map values by using hook_content_entity_map_values_alter().
+    // Support of importing a new entity type by providing mapping values in the hook.
+    $this->moduleHandler->alter('content_entity_map_values', $entity_type_id, $entity, $values);
+
     return $entity;
   }
