--- modules/autologout/src/AutologoutManager.php        2023-01-09 11:05:58
+++ modules/autologout/src/AutologoutManager.php   2023-01-09 11:05:59
@@ -185,7 +185,7 @@
    */
   public function inactivityMessage() {
     $message = Xss::filter($this->autoLogoutSettings->get('inactivity_message'));
-    $type = $this->autoLogoutSettings->get('inactivity_message_type');
+    $type = $this->autoLogoutSettings->get('inactivity_message_type') ?? '';
     if (!empty($message)) {
       $this->messenger->addMessage($this->t('@message', ['@message' => $message]), $type);
     }
