--- modules/perimeter/src/EventSubscriber/PerimeterSubscriber.php	2021-06-24 12:54:23.000000000 +0100
+++ modules/perimeter/src/EventSubscriber/PerimeterSubscriber.php	2021-06-24 12:54:23.000000000 +0100
@@ -13,6 +13,7 @@
 use Symfony\Component\HttpKernel\Event\ResponseEvent;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 use Symfony\Component\HttpKernel\KernelEvents;
+use Drupal\Core\Site\Settings;

 /**
  * On page not found events, ban the IP if the request is suspicious.
@@ -99,12 +100,16 @@
    * Ban the IP if the request is suspicious.
    */
   protected function handleBannedUrls(Request $request) {
+    $whitelist = Settings::get('reverse_proxy_addresses', []) ;
+    if (in_array($request->getClientIp(), $whitelist)){
+      return;
+    }
     if ($this->currentUser->hasPermission('bypass perimeter defence rules')) {
       return;
     }
     $request_path = $request->getPathInfo();
     $bannedPatterns = $this->configFactory->get('perimeter.settings')->get('not_found_exception_patterns');
-    if (!empty($bannedPatterns) && !empty($request_path)) {
+    if (!empty($bannedPatterns) && !empty($request_path) && (bool)\Drupal::state()->get('ibm_apim.ip_ban_enabled', true)) {
       foreach ($bannedPatterns as $pattern) {
         $pattern = trim($pattern);
         if (!empty($pattern) && preg_match($pattern, $request_path)) {
