--- modules/file_browser/file_browser.module	2018-08-09 11:14:45.000000000 +0100
+++ modules/file_browser/file_browser.module	2018-08-09 11:15:08.000000000 +0100
@@ -44,7 +44,7 @@
 
   if (isset($form['selection_display'])) {
     $form['selection_display']['#type'] = 'container';
-    $form['selection_display']['#attributes']['class'] = 'file-browser-actions';
+    $form['selection_display']['#attributes']['class'] = ['file-browser-actions'];
   }
 
   // Add a class for generic styling.
