--- modules/flood_unblock/src/Form/FloodUnblockAdminForm.php	2019-07-23 15:59:23.029693584 +0000
+++ modules/flood_unblock/src/Form/FloodUnblockAdminForm.php	2019-07-23 15:37:59.197064311 +0000
@@ -142,11 +142,11 @@
         $type = $form['table']['#options'][$value]['type'];
         switch ($type) {
           case 'ip':
-            $type = '.failed_login_ip';
+            $type = '%_ip';
             break;

           case 'user':
-            $type = '.failed_login_user';
+            $type = '%_user';
             break;

         }