--- modules/externalauth/src/ExternalAuth.php	2018-04-08 13:44:47.000000000 +0100
+++ modules/externalauth/src/ExternalAuth.php	2018-04-08 13:45:39.000000000 +0100
@@ -104,7 +104,7 @@
       unset($account_data['name']);
     }
     else {
-      $username = $provider . '_' . $authname;
+      $username = $authname;
     }

     $authmap_event = $this->eventDispatcher->dispatch(ExternalAuthEvents::AUTHMAP_ALTER, new ExternalAuthAuthmapAlterEvent($provider, $authname, $username, $authmap_data));
@@ -119,7 +119,7 @@
     $account_data = array_merge(
       [
         'name' => $authmap_event->getUsername(),
-        'init' => $provider . '_' . $authmap_event->getAuthname(),
+        'init' => $authmap_event->getAuthname(),
         'status' => 1,
         'access' => (int) $_SERVER['REQUEST_TIME'],
       ],
@@ -163,7 +163,7 @@
     // If a mapping (for the same provider) to this account already exists, we
     // silently skip saving this auth mapping.
     if (!$this->authmap->get($account->id(), $provider)) {
-      $authmap_event = $this->eventDispatcher->dispatch(new ExternalAuthAuthmapAlterEvent($provider, $authname, $account->getAccountName(), NULL), ExternalAuthEvents::AUTHMAP_ALTER);
+      $authmap_event = $this->eventDispatcher->dispatch(new ExternalAuthAuthmapAlterEvent($provider, $authname, $authname, NULL), ExternalAuthEvents::AUTHMAP_ALTER);
       $this->authmap->save($account, $provider, $authmap_event->getAuthname(), $authmap_event->getData());
     }
   }
