--- modules/password_policy/password_policy.module	2022-09-28 14:55:28.000000000 +0100
+++ modules/password_policy/password_policy.module	2022-09-29 14:35:00.000000000 +0100
@@ -10,6 +10,8 @@
 use Drupal\Core\Routing\RouteMatchInterface;
 use Drupal\Core\Session\AccountInterface;
 use Drupal\Core\Url;
+use Drupal\user\Entity\User;
+use Drupal\Core\Session\AccountProxyInterface;
 use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
 use Drupal\password_policy\Entity\PasswordPolicy;
 use Drupal\user\UserInterface;
@@ -27,6 +29,18 @@
   ];
 }

+// Used mainly for enforcing password policy username on password reset form
+function get_user(FormStateInterface $form_state) {
+  $user = $form_state->getFormObject()->getEntity();
+  $uid = $form_state->getValue('user_id');
+  if ((!isset($user) || $user->id() === 0) && isset($uid)) {
+    $user = User::load($uid);
+  }
+
+  return $user;
+
+}
+
 /**
  * Prepares variables for the password policy status.
  *
@@ -82,7 +96,7 @@
       '#constraints' => \Drupal::service('password_policy.validator')
         ->buildPasswordPolicyConstraintsTableRows(
           $form_state->getValue('pass', ''),
-          $form_state->getFormObject()->getEntity(),
+          get_user($form_state),
           _password_policy_get_edited_user_roles($form, $form_state)
       ),
       '#weight' => 4,
@@ -198,7 +212,7 @@
   // Check if user is authenticated externally.
   if (\Drupal::moduleHandler()->moduleExists('externalauth')) {
     $authmap = \Drupal::service('externalauth.authmap');
-    $account = $form_state->getFormObject()->getEntity();
+    $account = get_user($form_state);
     $external_ids = $authmap->getAll($account->id());
     if ($external_ids) {
       return;
@@ -217,18 +231,26 @@
   }
   $roles = _password_policy_get_edited_user_roles($form, $form_state);

-  $user = $form_state->getFormObject()->getEntity();
-  $user->setUsername($form_state->getValue('name', ''));
-  /** @var \Drupal\password_policy\PasswordPolicyValidationReport $validationReport */
-  $validationReport = \Drupal::service('password_policy.validator')->validatePassword(
-    $form_state->getValue('pass', ''),
-    $user,
-    $roles
-  );
+  $user = get_user($form_state);
+  if ($user instanceof User && $user->isNew()) {
+    $user->setUsername($form_state->getValue('name', ''));
+  }

-  if ($validationReport->isInvalid()) {
-    $form_state->setErrorByName('pass', t('The password does not satisfy the password policies.'));
-    $form_state->setError($form, $validationReport->getErrors());
+  if ($user instanceof AccountProxyInterface) {
+    $user = User::load($user->id());
+  }
+
+  if (!empty($form_state->getValue('pass')) && empty($form_state->getValue('pw_no_policy'))) {
+    $validationReport = \Drupal::service('password_policy.validator')->validatePassword(
+        $form_state->getValue('pass', ''),
+        $user,
+        $roles
+    );
+
+    if ($validationReport->isInvalid()) {
+        $form_state->setErrorByName('pass', t('The password does not satisfy the password policies.'));
+        $form_state->setError($form, $validationReport->getErrors());
+    }
   }
 }

@@ -239,7 +261,13 @@
   $roles = $form_state->getValue('roles');
   if (empty($roles)) {
     // Get if from $form; form state is always empty the first time.
-    $roles = $form['account']['roles']['#default_value'];
+    if(isset($form['roles']['#default_value'])) {
+      $roles = $form['roles']['#default_value'];
+    } else if(isset($form['account']['roles']['#default_value'])) {
+      $roles = $form['account']['roles']['#default_value'];
+    } else {
+      $roles = [];
+    }
   }
   $roles = (is_array($roles)) ? array_combine($roles, $roles) : [$roles];

@@ -274,7 +302,7 @@

   // Get User ID from User object.
   /** @var \Drupal\user\UserInterface $user */
-  $user = $form_state->getFormObject()->getEntity();
+  $user = get_user($form_state);
   $uid = $user->id();

   // Update if both current and new password fields are filled out. Depending
