--- modules/view_password/js/password.js	2023-01-27 19:04:57.475244610 +0000
+++ modules/view_password/js/password.js	2023-01-27 19:21:09.970208807 +0000
@@ -37,6 +37,12 @@
               .prop('type', 'text');
             $(this)
               .attr('aria-label', drupalSettings.view_password.hidePasswordLabel);
+            // autohide after 20s
+            setTimeout(function() {
+              $(this).addClass("eye-close").removeClass("eye-open");
+              $('.eye-close', context).prev(':text').prop('type', 'password');
+              $('button.shwpd').attr('aria-label', drupalSettings.view_password.showPasswordLabel);
+            }, 20000);
             if (icon_exposed_custom !== '') {
               $(this).css({'background-image': `url(${icon_exposed_custom})`});
             }
