--- modules/search404/src/Controller/Search404Controller.php	2022-11-07 10:38:26
+++ modules/search404/src/Controller/Search404Controller.php	2022-11-15 14:22:38
@@ -311,7 +311,7 @@
       if ($keys != '') {
         $custom_search_path = str_replace('@keys', $keys, $custom_search_path);
       }
-      return $this->search404Goto("/" . $custom_search_path);
+      return $this->search404Goto(base_path() . $custom_search_path);
     }

     if (empty($build)) {