--- modules/password_strength/src/Plugin/PasswordConstraint/PasswordStrength.php
+++ modules/password_strength/src/Plugin/PasswordConstraint/PasswordStrength.php
@@ -32,9 +32,8 @@ class PasswordStrength extends PasswordConstraintBase {
    * {@inheritdoc}
    */
   function validate($password, $user_context) {
-    unset($user_context['uid']);
 
-    $userData = array_values($user_context);
+    $userData = [$user_context->getUsername(), $user_context->getEmail()];
 
     $configuration = $this->getConfiguration();
     $validation = new PasswordPolicyValidation();
