--- modules/votingapi_widgets/src/Plugin/Field/FieldType/VotingApiField.php
+++ modules/votingapi_widgets/src/Plugin/Field/FieldType/VotingApiField.php
@@ -41,8 +41,8 @@ class VotingApiField extends FieldItemBase {
     return [
       'result_function' => 'vote_average',
       'widget_format' => 'fivestar',
-      'anonymous_window' => -1,
-      'user_window' => -1,
+      'anonymous_window' => -2,
+      'user_window' => -2,
     ] + parent::defaultFieldSettings();
   }

@@ -163,8 +163,9 @@ class VotingApiField extends FieldItemBase {
       $unit_options_form[$option] = $dateFormatter->formatInterval($option);
     }

-    $unit_options_form[0] = $this->t('never');
-    $unit_options_form[-1] = $this->t('votingapi default');
+    $unit_options_form[0] = $this->t('Immediately');
+    $unit_options_form[-1] = $this->t('Never');
+    $unit_options_form[-2] = $this->t('Votingapi default');

     $form['anonymous_window'] = [
       '#type' => 'select',
