--- modules/view_password/js/password.js	2020-10-12 14:13:51.000000000 +0100
+++ modules/view_password/js/password.js	2020-10-12 14:28:15.000000000 +0100
@@ -33,6 +33,11 @@
 
             if (className == 'eye-open') {
               $('.eye-open').prev(':password').prop('type', 'text');
+              // autohide after 20s
+              setTimeout(function() {
+                $('.shwpd').addClass("eye-close").removeClass("eye-open");
+                $('.eye-close').prev(':text').prop('type', 'password');
+              }, 20000);
             }
             else if (className == 'eye-close') {
               $('.eye-close').prev(':text').prop('type', 'password');
