--- modules/view_password/js/password.js	2023-01-27 19:04:57.475244610 +0000
+++ modules/view_password/js/password.js	2023-01-27 19:21:09.970208807 +0000
@@ -3,26 +3,35 @@
  * Contains \Drupal\view_password\password.js.
  */

-(function($, Drupal, drupalSettings, once) {
+(function ($, Drupal, drupalSettings, once) {
   'use strict';

   Drupal.behaviors.pwd = {
-    attach(context) {
+    attach: function (context) {
+      once('add-pwd-toggler', 'html').forEach(function () {
       var span_classes_custom = drupalSettings.view_password.span_classes || '';

-      $('.pwd-see :password', context).after(
-        `<button type="button" class="shwpd ${span_classes_custom} eye-close" aria-label="${drupalSettings.view_password.showPasswordLabel}"></button>`
-      );
-      $(once('view_password', '.shwpd', context))
-        .on('click', function() {
-          // To toggle the images.
-          $(this).toggleClass('eye-close eye-open');
+       $('.pwd-see :password', context).after(
+          `<button type="button" class="shwpd ${span_classes_custom} eye-close" aria-label="${drupalSettings.view_password.showPasswordLabel}"></button>`
+        );
+        $(once('view_password', '.shwpd', context))
+          .on('click', function () {
+            // To toggle the images.
+            $(this).toggleClass('eye-close eye-open');

-          if ($(this).hasClass('eye-open')) {
-            $('.eye-open', context)
-              .siblings(':password')
-              .prop('type', 'text');
+            if ($(this).hasClass('eye-open')) {
+
+              $('.eye-open', context)
+               .siblings(':password')
+               .prop('type', 'text');
+            // $('button.shwpd').attr('aria-label', drupalSettings.view_password.hidePasswordLabel);
             $('button.shwpd').attr('aria-label', drupalSettings.view_password.hidePasswordLabel);
+            // autohide after 20s
+            setTimeout(function() {
+              $('.shwpd').addClass("eye-close").removeClass("eye-open");
+              $('.eye-close', context).prev(':text').prop('type', 'password');
+              $('button.shwpd').attr('aria-label', drupalSettings.view_password.showPasswordLabel);
+            }, 20000);
           } else if ($(this).hasClass('eye-close')) {
             $('.eye-close', context)
               .siblings(':text')
@@ -30,6 +39,7 @@
             $('button.shwpd').attr('aria-label', drupalSettings.view_password.showPasswordLabel);
           }
         });
-    }
+      });
+    },
   };
 })(jQuery, Drupal, drupalSettings, once);
