--- modules/password_policy/password_policy.module	2018-03-15 14:21:38.940774953 +0000
+++ modules/password_policy/password_policy.module	2018-03-15 14:22:50.629147516 +0000
@@ -85,10 +85,6 @@
   $config = \Drupal::config('user.settings');
   $show_password_policy_status = TRUE;

-  if ($account->hasPermission('bypass password policies')) {
-    $show_password_policy_status = FALSE;
-  }
-
   if ($account->isAnonymous() and $config->get('verify_mail')) {
     $show_password_policy_status = FALSE;
   }
@@ -172,11 +168,15 @@
   $roles = $form_state->getValue('roles');
   if (empty($roles)) {
     // Get if from $form; form state is always empty the first time.
-    $roles = $form['account']['roles']['#default_value'];
+    if(isset($form['roles'])) {
+      $roles = $form['roles']['#default_value'];
+    } else if(isset($form['account']['roles'])) {
+      $roles = $form['account']['roles']['#default_value'];
+    }
   }
   $roles = array_combine($roles, $roles);

-  $original_roles = $form['account']['roles']['#default_value'];
+  $original_roles = (isset($form['roles'])) ? $form['roles']['#default_value'] : $form['account']['roles']['#default_value'];
   $original_roles = array_combine($original_roles, $original_roles);

   // Add user doesn't automatically register authenticated, so lets add it.
