--- modules/content_synchronizer/src/Base/JsonWriterTrait.php   2023-07-10 14:56:51
+++ modules/content_synchronizer/src/Base/JsonWriterTrait.php      2023-07-10 14:56:52
@@ -31,10 +31,11 @@
     $this->createDirectory($dir);

     $this->fileSystem()->prepareDirectory($dir, FileSystem::CHMOD_DIRECTORY);
-    $uri = $this->fileSystem()->saveData(Json::encode($data), $destination, FileSystem::EXISTS_REPLACE);
+    $uri = $this->fileSystem()->saveData(json_encode($data, JSON_UNESCAPED_UNICODE | JSON_PRETTY_PRINT), $destination, FileSystem::EXISTS_REPLACE);

     File::create([
       'uri' => $uri,
+      'status' => 0,
     ])->save();
   }