--- modules/cshs/build/jquery.simpler-select.js	2021-08-18 14:31:37.000000000 +0100
+++ modules/cshs/build/jquery.simpler-select.js	2021-08-18 14:32:53.000000000 +0100
@@ -151,7 +151,7 @@
         var $wrapper = $createElement('div') // Add level-specific class to ease the styling.
         .addClass([wrapperClass, wrapperClass + levelPrefix]) // Provide the read-only attribute for those who may need to query it.
         .attr('data-level', this.currentLevel);
-        var $select = $createElement('select').addClass(this.elementClasses).attr('id', selectId) // Suppress `Argument of type 'boolean' is not assignable to
+        var $select = $createElement('select').addClass(this.elementClasses).attr('id', selectId).attr('aria-labelledby', selectId + '-label') // Suppress `Argument of type 'boolean' is not assignable to
         // parameter of type 'string | number` because it's not true.
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
         // @ts-ignore TS2345.
@@ -324,4 +324,4 @@
 })(jQuery, 'select-wrapper', 'simplerSelect');
 /******/ })()
 ;
-//# sourceMappingURL=jquery.simpler-select.js.map
\ No newline at end of file
+//# sourceMappingURL=jquery.simpler-select.js.map
