--- modules/content_synchronizer/src/Service/EntityExportFormBuilder.php	2025-01-30 15:50:30
+++ modules/content_synchronizer/src/Service/EntityExportFormBuilder.php	2025-01-30 15:55:07
@@ -81,10 +81,20 @@
    * Add the export form in the entity edit form, if the entity is exportable.
    */
   public function addExportFields(array &$form, FormStateInterface $formState) {
+
     if ($this->isEntityEditForm($form, $formState)) {
-      $this->addExportFieldsToEntityForm($form, $formState);
+      $entity = $formState->getFormObject()->getEntity()->getEntityTypeId();
+      if (!$this->isBlocked($entity)) {
+        $this->addExportFieldsToEntityForm($form, $formState);
+      }
     }
   }
+
+  public function isBlocked(string $entity): bool {
+      $blockedTypes = [ "crop", "user", "api", "application", "consumerorg", "product", "event_log", "consumerorg_payment_method", "apic_app_application_subs", 
+      "apic_app_application_creds", "avatars_preview", "comment", "contact_message", "search_api_task", "vote", "vote_result" ];
+      return in_array($entity, $blockedTypes);
+    }
 
   /**
    * Return true if the form needs to have an export field.
