--- modules/clipboardjs/js/clipboard.js	2022-01-20 16:56:24.000000000 +0000
+++ modules/clipboardjs/js/clipboard.js	2022-01-20 16:59:02.000000000 +0000
@@ -20,7 +20,10 @@
       Drupal.clipboard.on('success', function (e) {
         let alertStyle = $(e.trigger).data('clipboardAlert');
         let alertText = $(e.trigger).data('clipboardAlertText');
-        let target = $(e.trigger).data('clipboardTarget');
+        let target = $(e.trigger.parentNode).find('input');
+        if (!$(target.length)) {
+          target = $(e.trigger).data('clipboardTarget');
+        }

         // Display as alert.
         if (alertStyle === 'alert') {
