--- modules/search404/src/Form/Search404Settings.php	2020-05-20 06:13:32.000000000 +0100
+++ modules/search404/src/Form/Search404Settings.php	2020-09-28 14:52:04.000000000 +0100
@@ -220,11 +220,11 @@
     if (!empty($form_state->getValue('search404_do_custom_search'))) {
       $custom_path = $form_state->getValue('search404_custom_search_path');
 
-      if (empty(preg_match("/\/@keys$/", $custom_path))) {
-        $form_state->setErrorByName('search404_custom_search_path', $this->t('Custom search path should be ends with search key pattern "/@keys".'));
+      if (empty(preg_match("/@keys$/", $custom_path))) {
+        $form_state->setErrorByName('search404_custom_search_path', $this->t('Custom search path should be ends with search key pattern "@keys".'));
       }
       $url_path = explode("@keys", $custom_path);
-      if (!empty(preg_match('/[\'^£!`$%&*()\{}\:.;,\[\]"@#~?><>,|=_+¬-]/', $url_path[0]))) {
+      if (!empty(preg_match('/[\'^£!`$%&*()\{}\:.;,\[\]"@#~><>,|_+¬-]/', $url_path[0]))) {
         $form_state->setErrorByName('search404_custom_search_path', t('Custom search path should not contains special characters other than "/"'));
       }
       if (strpos($custom_path, '/') === 0) {
