--- modules/check_dns/check_dns.module
+++ modules/check_dns/check_dns.module
@@ -42,7 +42,7 @@ 
  function check_dns_user_register_validate($form, FormStateInterface &$form_state) {
   $mail = $form_state->getValue('mail');
   $result = \Drupal::service('check_dns.service')->validateEmail($mail);
-  if (!$result) {
+  if (isset($mail) && !$result) {
     // If no record is found.
     $form_state->setErrorByName('mail', t('Your email domain is not recognised. Please enter a valid email id.'));
   }