--- modules/file_upload_secure_validator/src/Service/FileUploadSecureValidator.php	2020-11-19 09:08:51.000000000 +0000
+++ modules/file_upload_secure_validator/src/Service/FileUploadSecureValidator.php	2020-11-19 09:10:48.000000000 +0000
@@ -44,6 +44,27 @@
   ];
 
   /**
+   * A certificate mime descriptors equivalence group.
+   */
+  private $certMimetypes = [
+    'text/plain',
+    'application/pkix-cert',
+    'application/x-x509-ca-cert',
+    'application/x-pem-file',
+    'application/x-x509-user-cert',
+    'application/pgp-keys',
+    'application/pkcs8',
+    'application/pkcs10',
+    'application/pkix-crl',
+    'application/pkix-cert',
+    'application/x-pkcs12',
+    'application/x-pkcs7-certificates',
+    'application/x-pkcs7-certreqresp',
+    'application/x-pkcs7-crl',
+    'application/pkcs7-mime'
+  ];
+
+  /**
    * File validation function.
    *
    * @param \Drupal\file\Entity\File $file
@@ -71,6 +92,10 @@
     if (in_array($mime_by_filename, $this->xmlMimetypes) && in_array($mime_by_fileinfo, $this->xmlMimetypes)) {
       return [];
     }
+    // Exit when a certificate mime-type equivalence is found.
+    if (in_array($mime_by_filename, $this->certMimetypes) && in_array($mime_by_fileinfo, $this->certMimetypes)) {
+      return [];
+    }
     // Handle disagreement.
     if ($mime_by_filename !== $mime_by_fileinfo) {
       $errors[] = $this->t('There was a problem with this file. The uploaded file is of type @extension but the real content seems to be @real_extension', ['@extension' => $mime_by_filename, '@real_extension' => $mime_by_fileinfo]);
