--- modules/node_type_count/src/Controller/NodeTypeCountController.php	2019-06-04 13:48:56.000000000 +0100
+++ modules/node_type_count/src/Controller/NodeTypeCountController.php	2019-06-04 13:59:07.000000000 +0100
@@ -133,11 +133,15 @@
    * Count User Role.
    */
   public function userCountByRole($role_type_machine_name) {
-    $query = db_select('user__roles', 'ur')
-            ->fields('ur', array('entity_id'))
-            ->condition('roles_target_id', $role_type_machine_name);
-    $results = $query->countQuery()->execute()->fetchField();
-    return $results;
+    $user_storage = \Drupal::service('entity_type.manager')->getStorage('user');
+
+    $query = $user_storage->getQuery();
+    $query->condition('uid', 0, '<>');
+    if ($role_type_machine_name !== 'authenticated') {
+      $query->condition('roles', $role_type_machine_name);
+    }
+    $results = $query->execute();
+    return count($results);
   }
 
 }
