--- modules/workbench/src/Controller/WorkbenchContentController.php	2019-04-25 20:24:46.000000000 +0100
+++ modules/workbench/src/Controller/WorkbenchContentController.php	2019-04-25 20:25:06.000000000 +0100
@@ -4,11 +4,12 @@

 use Drupal\node\Controller\NodeController;
 use Drupal\views\Views;
+use Drupal\ibm_apim\Controller\IbmApimNodeController;

 /**
  * Generates the pages defined by Workbench.
  */
-class WorkbenchContentController extends NodeController {
+class WorkbenchContentController extends IbmApimNodeController {

   /**
    * Simple page to show list of content type to create.
