--- modules/captcha/image_captcha/js/image_captcha_refresh.js	2022-06-15 17:40:03.000000000 +0100
+++ modules/captcha/image_captcha/js/image_captcha_refresh.js	2022-06-27 15:41:16.000000000 +0100
@@ -22,7 +22,7 @@
         var $form = $(this).parents('form');
         // Send post query for getting new captcha data.
         var date = new Date();
-        var url = $(this).attr('href') + '?' + date.getTime();
+        var url = $(this).attr('href').replace(/^\/+/, '') + '?' + date.getTime();
         // Adding loader.
         $('.captcha').append('<div class="image_captcha_refresh_loader"></div>');
         $.get(
