--- modules/advagg/advagg_js_minify/src/Asset/JsMinifier.php	2019-06-27 11:43:00.000000000 +0100
+++ modules/advagg/advagg_js_minify/src/Asset/JsMinifier.php	2019-06-27 11:46:35.000000000 +0100
@@ -148,8 +148,8 @@
     // Check for JSMin errors.
     $error = jsmin_last_error_msg();
     if ($error != 'No error') {
-      $this->logger->warning('JSMin had an error processing, using JSqueeze fallback. Error details: ' . $error, []);
+      $this->logger->warning('JSMin had an error processing ' . $path . ', using JSqueeze fallback. Error details: ' . $error, []);
       $contents = $this->minifyJsqueeze($contents, $path);
       return;
     }
