--- ./modules/change_pwd_page/src/Form/ChangePasswordResetForm.php	2025-05-21 13:18:39.389722650 +0000
+++ ./modules/change_pwd_page/src/Form/ChangePasswordResetForm.php	2025-05-21 13:19:28.102722464 +0000
@@ -92,6 +92,9 @@
     $form['actions']['submit'] = [
       '#type' => 'submit',
       '#value' => $this->t('Log in'),
+      '#attributes' => [
+        'class' => ['button', 'button--primary', 'btn-primary', 'btn'],
+      ],
     ];
     return $form;
   }
