--- modules/session_limit/src/Services/SessionLimit.php	2019-10-30 14:32:31.000000000 +0000
+++ modules/session_limit/src/Services/SessionLimit.php	2019-10-30 14:33:31.000000000 +0000
@@ -15,7 +15,7 @@
 use Symfony\Component\HttpKernel\KernelEvents;
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Drupal\Core\Database\Connection;
-use Drupal\Core\Session\AccountProxy;
+use Drupal\Core\Session\AccountProxyInterface;
 use Drupal\Core\Session\SessionManager;
 use Drupal\Core\Extension\ModuleHandler;
 use Drupal\Core\Config\ConfigFactory;
@@ -113,7 +113,7 @@
    *   The event dispatcher service.
    * @param RouteMatchInterface $routeMatch
    *   The Route.
-   * @param AccountProxy $currentUser
+   * @param AccountProxyInterface $currentUser
    *   The current user.
    * @param SessionManager $sessionManager
    *   Session manager.
@@ -126,7 +126,7 @@
    * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $loggerFactory
    *   The logger factory.
    */
-  public function __construct(Connection $database, EventDispatcherInterface $eventDispatcher, RouteMatchInterface $routeMatch, AccountProxy $currentUser, SessionManager $sessionManager, ModuleHandler $moduleHandler, ConfigFactory $configFactory, MessengerInterface $messenger, LoggerChannelFactoryInterface $loggerFactory) {
+  public function __construct(Connection $database, EventDispatcherInterface $eventDispatcher, RouteMatchInterface $routeMatch, AccountProxyInterface $currentUser, SessionManager $sessionManager, ModuleHandler $moduleHandler, ConfigFactory $configFactory, MessengerInterface $messenger, LoggerChannelFactoryInterface $loggerFactory) {
     $this->routeMatch = $routeMatch;
     $this->database = $database;
     $this->eventDispatcher = $eventDispatcher;
@@ -525,7 +525,7 @@
    */
   public function getMessageSeverity() {
     return $this->configFactory->get('session_limit.settings')
-      ->get('session_limit_logged_out_message_severity');
+      ->get('session_limit_logged_out_message_severity') ?? '';
   }

   /**
