--- modules/panels/src/Form/LayoutPluginSelector.php
+++ modules/panels/src/Form/LayoutPluginSelector.php
@@ -104,6 +104,8 @@ class LayoutPluginSelector extends FormBase {
         'old_layout' => $variant_plugin->getConfiguration()['layout'],
         'new_layout' => $form_state->getValue('layout'),
       ];
+      $variant_plugin->setLayout($form_state->getValue('layout'), $form_state->getValue('layout_settings') ?: []);
+
       /** @var \Drupal\ctools\Wizard\EntityFormWizardInterface $wizard */
       $wizard = $form_state->getFormObject();
       $next_op = $wizard->getNextOp();
