diff -Naur modules/password_policy/src/Form/ConstraintDelete.php modules/password_policy/src/Form/ConstraintDelete.php
--- modules/password_policy/src/Form/ConstraintDelete.php
+++ modules/password_policy/src/Form/ConstraintDelete.php
@@ -6,7 +6,7 @@ use Drupal\Core\Form\ConfirmFormBase;
 use Drupal\Core\Form\ConfirmFormHelper;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Url;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -17,7 +17,7 @@ class ConstraintDelete extends ConfirmFormBase {
   /**
    * Temp store to maintain state between steps of the wizard.
    *
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstore;
 
@@ -46,7 +46,7 @@ class ConstraintDelete extends ConfirmFormBase {
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container) {
-    return new static($container->get('user.shared_tempstore'));
+    return new static($container->get('tempstore.shared'));
   }
 
   /**
diff -Naur modules/password_policy/src/Form/ConstraintEdit.php modules/password_policy/src/Form/ConstraintEdit.php
--- modules/password_policy/src/Form/ConstraintEdit.php
+++ modules/password_policy/src/Form/ConstraintEdit.php
@@ -9,7 +9,7 @@ use Drupal\Core\Ajax\RedirectCommand;
 use Drupal\Core\Form\FormBase;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Url;
-use Drupal\user\SharedTempStoreFactory;
+use Drupal\Core\TempStore\SharedTempStoreFactory;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -21,7 +21,7 @@ class ConstraintEdit extends FormBase {
   /**
    * Adding a tempstore for the multiple steps of the wizard form.
    *
-   * @var \Drupal\user\SharedTempStoreFactory
+   * @var \Drupal\Core\TempStore\SharedTempStoreFactory
    */
   protected $tempstore;
 
@@ -50,7 +50,7 @@ class ConstraintEdit extends FormBase {
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container) {
-    return new static($container->get('user.shared_tempstore'), $container->get('plugin.manager.password_policy.password_constraint'));
+    return new static($container->get('tempstore.shared'), $container->get('plugin.manager.password_policy.password_constraint'));
   }
 
   /**
