--- modules/password_policy/src/Entity/PasswordPolicy.php	2022-11-07 14:49:08
+++ modules/password_policy/src/Entity/PasswordPolicy.php	2023-08-24 16:03:08
@@ -71,7 +71,7 @@
    *
    * @var int
    */
-  protected $password_reset = 30;
+  protected $password_reset = 0; // This value has been overriden to a default value of 0 because we don't support password expire. 
 
   /**
    * Send email notification upon reset.
