--- modules/password_policy/src/Form/PasswordPolicyForm.php	2022-11-07 14:49:08
+++ modules/password_policy/src/Form/PasswordPolicyForm.php	2023-08-24 16:07:18
@@ -89,12 +89,14 @@
       '#title' => $this->t('Password Reset Days'),
       '#description' => $this->t('User password will reset after the selected number of days. 0 days indicates that passwords never expire.'),
       '#default_value' => $this->entity->getPasswordReset(),
+      '#disabled' => TRUE, // This entry is disabled because we don't support password expire
     ];
     $form['send_reset_email'] = [
       '#type' => 'checkbox',
       '#title' => $this->t('Send email upon password expiring'),
       '#description' => $this->t('If checked, an email will go to each user when their password expires, with a link to the request password reset email page.'),
       '#default_value' => $this->entity->getPasswordResetEmailValue(),
+      '#disabled' => TRUE, // This entry is disabled because we don't support password expire
     ];
 
     $form['send_pending_email'] = [
@@ -102,6 +104,7 @@
       '#title' => $this->t('Send pending email days before'),
       '#description' => $this->t('Send password expiration pending email X days before expiration. 0 days indicates this email will not be sent. The box above must also be checked. Separate by comma if sending multiple notifications.'),
       '#default_value' => implode(',', $this->entity->getPasswordPendingValue()),
+      '#disabled' => TRUE, // This entry is disabled because we don't support password expire
     ];
 
     $form['show_policy_table'] = [
