--- core/modules/system/system.module	2018-08-02 11:56:14.066705149 +0000
+++ core/modules/system/system.module	2018-08-02 12:35:07.444222208 +0000
@@ -155,7 +155,7 @@
       return $output;
 
     case 'system.modules_uninstall':
-      return '<p>' . t('The uninstall process removes all data related to a module.') . '</p>';
+      return '<p>' . t('The disable process removes all data related to a module.') . '</p>';
 
     case 'entity.block.edit_form':
       if (($block = $route_match->getParameter('block')) && $block->getPluginId() == 'system_powered_by_block') {

--- core/modules/system/src/Form/ModulesUninstallForm.php	2018-08-02 11:55:43.302432936 +0000
+++ core/modules/system/src/Form/ModulesUninstallForm.php	2018-08-02 12:37:24.645471095 +0000
@@ -131,7 +131,7 @@
 
       $form['uninstall'][$module->getName()] = [
         '#type' => 'checkbox',
-        '#title' => $this->t('Uninstall @module module', ['@module' => $name]),
+        '#title' => $this->t('Disable @module module', ['@module' => $name]),
         '#title_display' => 'invisible',
       ];
 
@@ -157,7 +157,7 @@
     $form['actions'] = ['#type' => 'actions'];
     $form['actions']['submit'] = [
       '#type' => 'submit',
-      '#value' => $this->t('Uninstall'),
+      '#value' => $this->t('Disable'),
     ];
 
     return $form;

--- core/modules/system/templates/system-modules-uninstall.html.twig	2018-08-02 13:04:27.951955222 +0000
+++ core/modules/system/templates/system-modules-uninstall.html.twig	2018-08-02 13:04:43.980099188 +0000
@@ -26,7 +26,7 @@
 <table class="responsive-enabled">
   <thead>
     <tr>
-      <th>{{ 'Uninstall'|t }}</th>
+      <th>{{ 'Disable'|t }}</th>
       <th>{{ 'Name'|t }}</th>
       <th>{{ 'Description'|t }}</th>
     </tr>

--- core/themes/stable/templates/admin/system-modules-uninstall.html.twig	2018-08-02 13:02:17.710786006 +0000
+++ core/themes/stable/templates/admin/system-modules-uninstall.html.twig	2018-08-02 13:02:32.170915760 +0000
@@ -24,7 +24,7 @@
 <table class="responsive-enabled" data-striping="1">
   <thead>
     <tr>
-      <th>{{ 'Uninstall'|t }}</th>
+      <th>{{ 'Disable'|t }}</th>
       <th>{{ 'Name'|t }}</th>
       <th>{{ 'Description'|t }}</th>
     </tr>

--- core/modules/system/src/Form/ModulesUninstallConfirmForm.php	2018-08-02 15:36:23.269811873 +0000
+++ core/modules/system/src/Form/ModulesUninstallConfirmForm.php	2018-08-02 15:39:12.999331799 +0000
@@ -90,14 +90,14 @@
    * {@inheritdoc}
    */
   public function getQuestion() {
-    return $this->t('Confirm uninstall');
+    return $this->t('Confirm disable');
   }
 
   /**
    * {@inheritdoc}
    */
   public function getConfirmText() {
-    return $this->t('Uninstall');
+    return $this->t('Disable');
   }
 
   /**
@@ -111,7 +111,7 @@
    * {@inheritdoc}
    */
   public function getDescription() {
-    return $this->t('Would you like to continue with uninstalling the above?');
+    return $this->t('Would you like to continue with disabling the above?');
   }
 
   /**
@@ -131,12 +131,12 @@
 
     // Prevent this page from showing when the module list is empty.
     if (empty($this->modules)) {
-      drupal_set_message($this->t('The selected modules could not be uninstalled, either due to a website problem or due to the uninstall confirmation form timing out. Please try again.'), 'error');
+      drupal_set_message($this->t('The selected modules could not be disabled, either due to a website problem or due to the disable confirmation form timing out. Please try again.'), 'error');
       return $this->redirect('system.modules_uninstall');
     }
 
     $data = system_rebuild_module_data();
-    $form['text']['#markup'] = '<p>' . $this->t('The following modules will be completely uninstalled from your site, and <em>all data from these modules will be lost</em>!') . '</p>';
+    $form['text']['#markup'] = '<p>' . $this->t('The following modules will be disabled for your site, and <em>all data from these modules will be lost</em>!') . '</p>';
     $form['modules'] = [
       '#theme' => 'item_list',
       '#items' => array_map(function ($module) use ($data) {
@@ -161,7 +161,7 @@
     // Uninstall the modules.
     $this->moduleInstaller->uninstall($this->modules);

-    drupal_set_message($this->t('The selected modules have been uninstalled.'));
+    drupal_set_message($this->t('The selected modules have been disabled.'));
     $form_state->setRedirectUrl($this->getCancelUrl());
   }

