--- core/scripts/run-tests.sh	2017-01-20 11:53:01.086840157 +0000
+++ core/scripts/run-tests.sh	2017-01-20 11:58:35.180762897 +0000
@@ -473,6 +473,8 @@
     }
   }
 
+  simpletest_script_get_db_values($args['url']);
+
   chdir(realpath(__DIR__ . '/../..'));
 }
 
@@ -609,6 +611,52 @@
   }
 }
 
+/**
+ * Read in the database values from the nginx config file
+ * @param $url
+ */
+function simpletest_script_get_db_values($url) {
+  $pathstring = '';
+  $url_parts = parse_url($url);
+  if (isset($url_parts['path'])) {
+    $pathstring = str_replace('/', '.', $url_parts['path']);
+    $pathstring = ltrim($pathstring, ".");
+    $pathstring = $pathstring . '.';
+  }
+  $sitename = $pathstring . $url_parts['host'];
+  $filename = '/var/aegir/config/server_master/nginx/vhost.d/' . $sitename;
+  if (file_exists($filename)) {
+    $txt_file = file_get_contents($filename);
+    $rows = explode("\n", $txt_file);
+    foreach ($rows as $row => $data) {
+      $row_parts = preg_split('/\s+/', $data);
+      if (isset($row_parts[1]) && $row_parts[1] == 'fastcgi_param' && isset($row_parts[2]) && isset($row_parts[3])) {
+        $row_parts[3] = rtrim($row_parts[3], ';');
+        switch ($row_parts[2]) {
+          case 'db_type':
+            $_SERVER['db_type'] = $row_parts[3];
+            break;
+          case 'db_name':
+            $_SERVER['db_name'] = $row_parts[3];
+            break;
+          case 'db_user':
+            $_SERVER['db_user'] = $row_parts[3];
+            break;
+          case 'db_passwd':
+            $_SERVER['db_passwd'] = $row_parts[3];
+            break;
+          case 'db_host':
+            $_SERVER['db_host'] = $row_parts[3];
+            break;
+          case 'db_port':
+            $_SERVER['db_port'] = $row_parts[3];
+            break;
+        }
+      }
+    }
+  }
+}
+
 /**
  * Execute a batch of tests.
  */
