--- modules/search_api/modules/search_api_db/src/DatabaseCompatibility/MySql.php	2023-10-15 11:30:01
+++ modules/search_api/modules/search_api_db/src/DatabaseCompatibility/MySql.php	2023-11-02 14:22:11
@@ -47,10 +47,7 @@
     // As MySQL removes trailing whitespace when computing primary keys, we need
     // to do the same or pseudo-duplicates could cause an exception ("Integrity
     // constraint violation: Duplicate entry") during indexing.
-    if ($type !== 'text') {
-      $value = rtrim($value);
-    }
-    return $value;
+    return rtrim($value);
   }

   /**

--- modules/search_api/modules/search_api_db/src/Plugin/search_api/backend/Database.php 2023-10-15 11:30:01
+++ modules/search_api/modules/search_api_db/src/Plugin/search_api/backend/Database.php    2023-11-02 14:24:05
@@ -1384,7 +1384,7 @@
       if ($this->configuration['phrase'] === 'bigram') {
         // Now add a bigram for this word and the last. In case this is the
         // first word, there is no bigram to add.
-        if ($prev_word === NULL) {
+        if ($prev_word === NULL || mb_strlen($prev_word) + 1 >= static::TOKEN_LENGTH_MAX) {
           continue;
         }
