--- vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/SendmailTransport.php	2018-03-26 16:07:27.000000000 +0100
+++ vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/SendmailTransport.php	2018-03-26 16:07:59.000000000 +0100
@@ -118,11 +118,12 @@
 
             $buffer->initialize(array_merge($this->_params, array('command' => $command)));
 
-            if (false === strpos($command, ' -i') && false === strpos($command, ' -oi')) {
-                $buffer->setWriteTranslations(array("\r\n" => "\n", "\n." => "\n.."));
-            } else {
+            // APIC commented out to avoid .. in long URLs
+            //if (false === strpos($command, ' -i') && false === strpos($command, ' -oi')) {
+            //    $buffer->setWriteTranslations(array("\r\n" => "\n", "\n." => "\n.."));
+            //} else {
                 $buffer->setWriteTranslations(array("\r\n" => "\n"));
-            }
+            //}
 
             $count = count((array) $message->getTo())
                 + count((array) $message->getCc())
