--- vendor/symfony/http-foundation/Request.php  2024-07-26 13:41:01
+++ vendor/symfony/http-foundation/Request.php     2024-09-03 15:07:38
@@ -1063,9 +1063,15 @@
      */
     public function getHost(): string
     {
-        if ($this->isFromTrustedProxy() && $host = $this->getTrustedValues(self::HEADER_X_FORWARDED_HOST)) {
-            $host = $host[0];
-        } elseif (!$host = $this->headers->get('HOST')) {
+        // velox/devportal#7594 - we have commented the code related to X_FORWARDED_HOST and trusted proxies here
+        // as this creates issues in OCP environments where sometimes getTrustedValues() returns values which
+        // cause us to use the wrong hostname to search for a cookie which won't exist. This then means authorize.php
+        // will reject calls such as UI theme upload as it can't find the cookie to authenticate the user.
+
+        //if ($this->isFromTrustedProxy() && $host = $this->getTrustedValues(self::HEADER_X_FORWARDED_HOST)) {
+        //    $host = $host[0];
+        //} elseif (!$host = $this->headers->get('HOST')) {
+        if (!$host = $this->headers->get('HOST')) {
             if (!$host = $this->server->get('SERVER_NAME')) {
                 $host = $this->server->get('SERVER_ADDR', '');
             }
