--- vendor/composer/composer/src/Composer/Util/TlsHelper.php	2024-06-10 21:11:12.000000000 +0100
+++ vendor/composer/composer/src/Composer/Util/TlsHelper.php	2024-07-23 11:06:59.550431933 +0100
@@ -150,7 +150,7 @@
         $pemtrim = substr($pubkeypem, strpos($pubkeypem, $start) + strlen($start), (strlen($pubkeypem) - strpos($pubkeypem, $end)) * (-1));
         $der = base64_decode($pemtrim);

-        return hash('sha1', $der);
+        return openssl_digest($der , 'sha1');
     }

     /**
