--- vendor/drush/drush/drush.php        2024-03-01 19:50:19
+++ vendor/drush/drush/drush.php   2024-03-01 19:50:15
@@ -136,6 +136,26 @@
 $di = new DependencyInjection();
 $di->desiredHandlers(['errorHandler', 'shutdownHandler']);
 $runtime = new Runtime($preflight, $di);
+foreach ($_SERVER['argv'] as $param) {
+    if (str_starts_with($param, '--uri')) {
+        # Get the database creds from the vhost file
+        $filename = "/var/aegir/config/server_master/nginx/vhost.d/" . explode('=', $param)[1];
+        if (is_file($filename)) {
+            $file = new SplFileObject($filename);
+
+            while (!$file->eof()) {
+                $line=rtrim($file->fgets());
+                if (strpos($line,"  fastcgi_param db_") === 0) {
+                    $line = ltrim($line, "  fastcgi_param ");
+                    $line = rtrim($line, ";");
+                    $bits = array_values(array_filter(explode(' ',$line)));
+                    $_SERVER[$bits[0]] = $bits[1];
+                }
+            }
+        }
+    }
+}
+
 $status_code = $runtime->run($_SERVER['argv']);

 exit($status_code);
