--- vendor/behat/mink-zombie-driver/bin/mink-zombie-server.js	2017-05-22 12:43:51.000000000 +0100
+++ vendor/behat/mink-zombie-driver/bin/mink-zombie-server.js	2017-05-22 12:44:03.000000000 +0100
@@ -1,16 +1,13 @@
 #!/usr/bin/env node
 var net = require('net');
 var zombie = require('zombie');
-var Tough = require('zombie/node_modules/tough-cookie');
 var browser = null;
 var pointers = [];
 var buffer = '';
 var host = process.env.HOST || '127.0.0.1';
 var port = process.env.PORT || 8124;
 
-Tough.Cookie.prototype.cookieString = function cookieString() {
-    return this.key + '=' + (this.value == null ? '' : this.value);
-};
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
 
 var zombieVersionCompare = function (v2, op) {
     var version_compare = function (v1, v2, operator) {
@@ -113,7 +110,7 @@
 
     stream.on('end', function () {
         if (browser == null) {
-            browser = new zombie();
+            browser = new zombie('{"waitDuration":"30s", "strictSSL":"false"}');
 
             // Clean up old pointers
             pointers = [];

