--- modules/votingapi_widgets/src/FieldVoteResultBase.php       2024-09-20 16:40:15
+++ modules/votingapi_widgets/src/FieldVoteResultBase.php  2024-09-20 16:40:41
@@ -26,7 +26,7 @@
   /**
    * Calculate results.
    */
-  public function calculateResult($votes) {
+  public function calculateResult(array $votes): float {
     return count($votes);
   }

--- modules/votingapi_widgets/src/Plugin/VoteResultFunction/FieldAverage.php    2024-09-20 16:41:20
+++ modules/votingapi_widgets/src/Plugin/VoteResultFunction/FieldAverage.php       2024-09-20 16:41:17
@@ -28,7 +28,7 @@
   /**
    * {@inheritdoc}
    */
-  public function calculateResult($votes) {
+  public function calculateResult(array $votes): float {
     $total = 0;
     $votes = $this->getVotesForField($votes);
     foreach ($votes as $vote) {

--- modules/votingapi_widgets/src/Plugin/VoteResultFunction/FieldCount.php      2024-09-20 16:41:51
+++ modules/votingapi_widgets/src/Plugin/VoteResultFunction/FieldCount.php 2024-09-20 16:41:49
@@ -28,7 +28,7 @@
   /**
    * {@inheritdoc}
    */
-  public function calculateResult($votes) {
+  public function calculateResult(array $votes): float {
     $votes = $this->getVotesForField($votes);
     return count($votes);
   }

--- modules/votingapi_widgets/src/Plugin/VoteResultFunction/FieldUseful.php     2024-09-19 15:45:04
+++ modules/votingapi_widgets/src/Plugin/VoteResultFunction/FieldUseful.php        2024-09-20 16:42:22
@@ -28,7 +28,7 @@
   /**
    * {@inheritdoc}
    */
-  public function calculateResult($votes) {
+  public function calculateResult(array $votes): float {
     $total = 0;
     $votes = $this->getVotesForField($votes);
     foreach ($votes as $vote) {
