--- /var/aegir/.drush/provision/Provision/Config/Drupal/provision_drupal_settings_8.tpl.php.orig	2020-07-30 09:35:26.221984798 +0000
+++ /var/aegir/.drush/provision/Provision/Config/Drupal/provision_drupal_settings_8.tpl.php	2020-07-30 09:35:36.514032876 +0000
@@ -48,13 +48,6 @@
      * should probably be fixed in Drush.
      */
     'port' => (string) $_SERVER['db_port'],
-    'pdo' => array (
-        PDO::MYSQL_ATTR_SSL_KEY => '/etc/secrets/client/tls.key',
-        PDO::MYSQL_ATTR_SSL_CERT => '/etc/secrets/client/tls.crt',
-        PDO::MYSQL_ATTR_SSL_CA => '/etc/secrets/ca/tls.crt',
-        PDO::MYSQL_ATTR_SSL_CIPHER => 'ECDHE-RSA-AES256-GCM-SHA384',
-        PDO::MYSQL_ATTR_SSL_VERIFY_SERVER_CERT => false
-    ),
 
 <?php if ($utf8mb4_is_configurable && $utf8mb4_is_supported): ?>
     'charset' => 'utf8mb4',
